diff -Naur gcc-6.2.0.orig/gcc/config/linux.c gcc-6.2.0/gcc/config/linux.c
--- gcc-6.2.0.orig/gcc/config/linux.c	2016-12-24 15:55:23.308342638 -0800
+++ gcc-6.2.0/gcc/config/linux.c	2016-12-24 16:00:51.852488459 -0800
@@ -26,7 +26,7 @@
 bool
 linux_libc_has_function (enum function_class fn_class)
 {
-  if (OPTION_GLIBC)
+  if (OPTION_GLIBC || OPTION_MUSL)
     return true;
   if (OPTION_BIONIC)
     if (fn_class == function_c94
diff -Naur gcc-6.2.0.orig/gcc/config.gcc gcc-6.2.0/gcc/config.gcc
--- gcc-6.2.0.orig/gcc/config.gcc	2016-12-24 15:55:15.957272251 -0800
+++ gcc-6.2.0/gcc/config.gcc	2016-12-24 16:02:12.297258721 -0800
@@ -1484,7 +1484,7 @@
 		extra_options="${extra_options} linux-android.opt"
 		# Assume modern glibc if not targeting Android nor uclibc.
 		case ${target} in
-		*-*-*android*|*-*-*uclibc*)
+		*-*-*android*|*-*-*uclibc*|*-*-*musl*)
 		  ;;
 		*)
 		  default_gnu_indirect_function=yes
@@ -1553,7 +1553,7 @@
 		extra_options="${extra_options} linux-android.opt"
 		# Assume modern glibc if not targeting Android nor uclibc.
 		case ${target} in
-		*-*-*android*|*-*-*uclibc*)
+		*-*-*android*|*-*-*uclibc*|*-*-*musl*)
 		  ;;
 		*)
 		  default_gnu_indirect_function=yes
