diff --git a/gnulib_m4/signbit.m4 b/gnulib_m4/signbit.m4
index 9e7884d..ef1ed6f 100644
--- a/gnulib_m4/signbit.m4
+++ b/gnulib_m4/signbit.m4
@@ -29,12 +29,12 @@ AC_DEFUN([gl_SIGNBIT],
         [gl_cv_func_signbit=yes],
         [gl_cv_func_signbit=no],
         [case "$host_os" in
-                          # Guess yes on glibc systems.
-           *-gnu* | gnu*) gl_cv_func_signbit="guessing yes" ;;
-                          # Guess yes on native Windows.
-           mingw*)        gl_cv_func_signbit="guessing yes" ;;
-                          # If we don't know, assume the worst.
-           *)             gl_cv_func_signbit="guessing no" ;;
+				    # Guess yes on glibc systems.
+           *-gnu* | gnu* | *musl*) gl_cv_func_signbit="guessing yes" ;;
+                           	    # Guess yes on native Windows.
+           mingw*)        	    gl_cv_func_signbit="guessing yes" ;;
+                          	    # If we don't know, assume the worst.
+           *)             	    gl_cv_func_signbit="guessing no" ;;
          esac
         ])
     ])
@@ -60,17 +60,17 @@ AC_DEFUN([gl_SIGNBIT],
         [gl_cv_func_signbit_gcc=yes],
         [gl_cv_func_signbit_gcc=no],
         [case "$host_os" in
-                          # Guess yes on glibc systems.
-           *-gnu* | gnu*) gl_cv_func_signbit_gcc="guessing yes" ;;
-                          # Guess yes on mingw, no on MSVC.
-           mingw*)        if test -n "$GCC"; then
-                            gl_cv_func_signbit_gcc="guessing yes"
-                          else
-                            gl_cv_func_signbit_gcc="guessing no"
-                          fi
-                          ;;
-                          # If we don't know, assume the worst.
-           *)             gl_cv_func_signbit_gcc="guessing no" ;;
+				    # Guess yes on glibc systems.
+           *-gnu* | gnu* | *musl*) gl_cv_func_signbit_gcc="guessing yes" ;;
+                           	    # Guess yes on mingw, no on MSVC.
+           mingw*)        	    if test -n "$GCC"; then
+                            	       gl_cv_func_signbit_gcc="guessing yes"
+                          	    else
+				       gl_cv_func_signbit_gcc="guessing no"
+                          	    fi
+                          	    ;;
+                          	    # If we don't know, assume the worst.
+           *)             	    gl_cv_func_signbit_gcc="guessing no" ;;
          esac
         ])
     ])
