--- libnfsidmap-0.27/configure.ac.orig	2018-05-23 00:07:14.273119199 -0700
+++ libnfsidmap-0.27/configure.ac	2018-05-23 00:07:24.285190437 -0700
@@ -14,7 +14,7 @@
 
 # Checks for libraries.
 
-AC_CHECK_LIB([resolv], [__res_querydomain], , AC_MSG_ERROR(res_querydomain needed))
+AC_CHECK_LIB([resolv], [res_querydomain], , AC_MSG_ERROR(res_querydomain needed))
 
 AC_ARG_ENABLE([ldap],
 	[AS_HELP_STRING([--disable-ldap],[Disable support for LDAP @<:@default=detect@:>@])])
