diff -Naur mesa-18.0.0-rc2.orig/src/gallium/state_trackers/nine/nine_debug.c mesa-18.0.0-rc2/src/gallium/state_trackers/nine/nine_debug.c
--- mesa-18.0.0-rc2.orig/src/gallium/state_trackers/nine/nine_debug.c	2018-01-24 13:20:13.287944874 -0800
+++ mesa-18.0.0-rc2/src/gallium/state_trackers/nine/nine_debug.c	2018-01-24 13:23:34.722941267 -0800
@@ -73,8 +73,8 @@
     }
 
 #if defined(HAVE_PTHREAD)
-#  if defined(__GNU_LIBRARY__) && defined(__GLIBC__) && defined(__GLIBC_MINOR__) && \
-      (__GLIBC__ >= 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 12))
+#  if defined(__linux__) && !(defined(__GLIBC__) || \
+      (__GLIBC__ < 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 12)))
     if (dbg_flags & DBG_TID)
         tid = pthread_self();
 #  endif
diff -Naur mesa-18.0.0-rc2.orig/src/util/u_thread.h mesa-18.0.0-rc2/src/util/u_thread.h
--- mesa-18.0.0-rc2.orig/src/util/u_thread.h	2018-01-24 13:20:13.166944876 -0800
+++ mesa-18.0.0-rc2/src/util/u_thread.h	2018-01-24 13:24:11.028940616 -0800
@@ -61,9 +61,8 @@
 static inline void u_thread_setname( const char *name )
 {
 #if defined(HAVE_PTHREAD)
-#  if defined(__GNU_LIBRARY__) && defined(__GLIBC__) && defined(__GLIBC_MINOR__) && \
-      (__GLIBC__ >= 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 12)) && \
-      defined(__linux__)
+#  if defined(__linux__) && !(defined(__GLIBC__) || \
+      (__GLIBC__ < 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 12)))
    pthread_setname_np(pthread_self(), name);
 #  endif
 #endif
@@ -93,8 +92,8 @@
 static inline bool u_thread_is_self(thrd_t thread)
 {
 #if defined(HAVE_PTHREAD)
-#  if defined(__GNU_LIBRARY__) && defined(__GLIBC__) && defined(__GLIBC_MINOR__) && \
-      (__GLIBC__ >= 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 12))
+#  if defined(__linux__) && !(defined(__GLIBC__) || \
+      (__GLIBC__ < 3 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 12)))
    return pthread_equal(pthread_self(), thread);
 #  endif
 #endif
