diff -Naur mesa-17.3.1.orig/src/util/u_endian.h mesa-17.3.1/src/util/u_endian.h
--- mesa-17.3.1.orig/src/util/u_endian.h	2017-12-29 14:19:33.532915798 -0800
+++ mesa-17.3.1/src/util/u_endian.h	2017-12-29 14:21:55.547913255 -0800
@@ -27,7 +27,7 @@
 #ifndef U_ENDIAN_H
 #define U_ENDIAN_H
 
-#if defined(__GLIBC__) || defined(ANDROID) || defined(__CYGWIN__)
+#if defined(__linux__) || defined(__CYGWIN__)
 #include <endian.h>
 
 #if __BYTE_ORDER == __LITTLE_ENDIAN
