Index: radvd-2.17/includes.h
===================================================================
--- radvd-2.17.orig/includes.h
+++ radvd-2.17/includes.h
@@ -76,7 +76,12 @@
 #include <sys/sysctl.h>
 #endif
 
+#if !defined(__GLIBC__) && defined(linux)
+#include <linux/if.h>
+#define IF_NAMESIZE IFNAMSIZ
+#else
 #include <net/if.h>
+#endif
 
 #ifdef HAVE_NET_IF_DL_H
 #include <net/if_dl.h>
Index: radvd-2.17/netlink.c
===================================================================
--- radvd-2.17.orig/netlink.c
+++ radvd-2.17/netlink.c
@@ -22,7 +22,14 @@
 #include <errno.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+
+#if !defined(__GLIBC__) && defined(linux)
+#include <linux/if.h>
+#define IF_NAMESIZE IFNAMSIZ
+#else
 #include <net/if.h>
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
