diff -Naur appstream-glib-0.7.9.orig/client/as-util.c appstream-glib-0.7.9/client/as-util.c
--- appstream-glib-0.7.9.orig/client/as-util.c	2018-06-29 14:38:14.133404736 -0700
+++ appstream-glib-0.7.9/client/as-util.c	2018-06-29 14:38:54.489404014 -0700
@@ -26,6 +26,7 @@
 #include <glib-unix.h>
 #include <gio/gio.h>
 
+#include <stdint.h>
 #include <appstream-glib.h>
 #include <archive_entry.h>
 #include <archive.h>
diff -Naur appstream-glib-0.7.9.orig/libappstream-glib/as-utils.c appstream-glib-0.7.9/libappstream-glib/as-utils.c
--- appstream-glib-0.7.9.orig/libappstream-glib/as-utils.c	2018-06-29 14:38:14.135404736 -0700
+++ appstream-glib-0.7.9/libappstream-glib/as-utils.c	2018-06-29 14:38:37.927404310 -0700
@@ -33,6 +33,7 @@
 #include "config.h"
 
 #include <fnmatch.h>
+#include <stdint.h>
 #include <string.h>
 #include <archive_entry.h>
 #include <archive.h>
