diff -Naur efivar-27.orig/src/linux.c efivar-27/src/linux.c
--- efivar-27.orig/src/linux.c	2016-08-19 12:28:16.782365757 -0700
+++ efivar-27/src/linux.c	2016-08-19 12:30:12.852460305 -0700
@@ -48,6 +48,15 @@
 #include "linux.h"
 #include "util.h"
 
+#ifndef strndupa
+#define strndupa(s, n) \
+       (__extension__ ({const char *__in = (s); \
+                        size_t __len = strnlen (__in, (n)) + 1; \
+                        char *__out = (char *) alloca (__len); \
+                        __out[__len-1] = '\0'; \
+                        (char *) memcpy (__out, __in, __len-1);}))
+#endif
+
 int
 __attribute__((__visibility__ ("hidden")))
 set_disk_and_part_name(struct disk_info *info)
