diff --git a/src/libreplace/arpa_nameser.h b/src/libreplace/arpa_nameser.h
index e59ca3d..35e32b3 100644
--- a/src/libreplace/arpa_nameser.h
+++ b/src/libreplace/arpa_nameser.h
@@ -59,11 +59,6 @@
 #  define __P(x) x
 #endif
 
-#if defined( HAVE_ARPA_NAMESER_H ) && defined( HAVE_NS_TYPE )
-#define dn_expand		__dn_expand
-int		dn_expand __P((const u_char *, const u_char *, const u_char *,
-			       char *, int));
-#endif
 #define dn_skipname		__dn_skipname
 int		dn_skipname __P((const u_char *, const u_char *));
 
