diff -Naur gpm-1.20.7.orig/src/daemon/open_console.c gpm-1.20.7/src/daemon/open_console.c
--- gpm-1.20.7.orig/src/daemon/open_console.c	2012-10-26 21:21:38.000000000 +0000
+++ gpm-1.20.7/src/daemon/open_console.c	2015-07-20 19:31:22.225131984 +0000
@@ -22,6 +22,7 @@
 #include <fcntl.h>                  /* open and co.      */
 #include <sys/stat.h>               /* stat()            */
 #include <sys/ioctl.h>              /* ioctl             */
+#include <sys/sysmacros.h>          /* major             */
 
 /* Linux specific (to be outsourced in gpm2 */
 #include <linux/serial.h>           /* for serial console check */
diff -Naur gpm-1.20.7.orig/src/prog/display-buttons.c gpm-1.20.7/src/prog/display-buttons.c
--- gpm-1.20.7.orig/src/prog/display-buttons.c	2012-10-26 21:21:38.000000000 +0000
+++ gpm-1.20.7/src/prog/display-buttons.c	2015-07-20 19:27:48.489140025 +0000
@@ -36,6 +36,7 @@
 #include <stdio.h>            /* printf()             */
 #include <time.h>             /* time()               */
 #include <errno.h>            /* errno                */
+#include <sys/select.h>       /* fd_set, select, FD_* */
 #include <gpm.h>              /* gpm information      */
 
 /* display resulting data */
diff -Naur gpm-1.20.7.orig/src/prog/display-coords.c gpm-1.20.7/src/prog/display-coords.c
--- gpm-1.20.7.orig/src/prog/display-coords.c	2012-10-26 21:21:38.000000000 +0000
+++ gpm-1.20.7/src/prog/display-coords.c	2015-07-20 19:28:45.829137867 +0000
@@ -37,6 +37,7 @@
 #include <stdio.h>            /* printf()             */
 #include <time.h>             /* time()               */
 #include <errno.h>            /* errno                */
+#include <sys/select.h>       /* fd_set, select, FD_* */
 #include <gpm.h>              /* gpm information      */
 
 /* display resulting data */
