From 2c2d8d71838e9de7f711ea753414301e02f1e5a8 Mon Sep 17 00:00:00 2001
From: 
Date: Sat, 4 Sep 2021 10:08:49 -0500
Subject: [PATCH] Fix uninitialized mnebuf

---
 libcpu/i386_disasm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libcpu/i386_disasm.c b/libcpu/i386_disasm.c
index fd7340c..61a8256 100644
--- a/libcpu/i386_disasm.c
+++ b/libcpu/i386_disasm.c
@@ -714,6 +714,7 @@ i386_disasm (Ebl *ebl __attribute__((unused)),
 
 		case 'm':
 		  /* Mnemonic.  */
+		  str = mnebuf;
 
 		  if (unlikely (instrtab[cnt].mnemonic == MNE_INVALID))
 		    {
-- 
2.32.0

