From 9e10d040b12bd0aa5c4c95848763426cd9aaf089 Mon Sep 17 00:00:00 2001
From: 
Date: Thu, 3 Feb 2022 18:27:20 -0600
Subject: [PATCH] check for musl

---
 configure.ac | 23 ++++-------------------
 1 file changed, 4 insertions(+), 19 deletions(-)

diff --git a/configure.ac b/configure.ac
index 092fb04cc..ce4e408a6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -879,7 +879,7 @@ solaris*)
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*)
+linux-gnu*|linux-musl*|k*bsd*-gnu*)
     using_freetype_fontconfig=yes
     using_x11=yes
     build_skia=yes
@@ -4933,7 +4933,7 @@ kfreebsd*)
     esac
     ;;
 
-linux-gnu*)
+linux-gnu*|linux-musl*)
     COM=GCC
     OS=LINUX
     RTL_OS=Linux
@@ -8593,7 +8593,7 @@ if test -n "$ENABLE_JAVA" -a -z "$JAVAINC"; then
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linux-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -11235,21 +11235,6 @@ AC_SUBST(SYSTEM_LIBNUMBERTEXT)
 AC_SUBST(SYSTEM_LIBNUMBERTEXT_DATA)
 AC_SUBST(LIBNUMBERTEXT_CFLAGS)
 
-dnl ***************************************
-dnl testing libc version for Linux...
-dnl ***************************************
-if test "$_os" = "Linux"; then
-    AC_MSG_CHECKING([whether libc is >= 2.1.1])
-    exec 6>/dev/null # no output
-    AC_CHECK_LIB(c, gnu_get_libc_version, HAVE_LIBC=yes; export HAVE_LIBC)
-    exec 6>&1 # output on again
-    if test "$HAVE_LIBC"; then
-        AC_MSG_RESULT([yes])
-    else
-        AC_MSG_ERROR([no, upgrade libc])
-    fi
-fi
-
 dnl =========================================
 dnl Check for uuidgen
 dnl =========================================
@@ -14221,7 +14206,7 @@ else
 
     case "$host_os" in
 
-    aix*|dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    aix*|dragonfly*|freebsd*|linux-gnu*|linux-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != ""; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
-- 
2.34.1

