diff -Naur elfutils-0.171.orig/lib/error.h elfutils-0.171/lib/error.h
--- elfutils-0.171.orig/lib/error.h	1969-12-31 16:00:00.000000000 -0800
+++ elfutils-0.171/lib/error.h	2018-06-11 10:11:15.910181114 -0700
@@ -0,0 +1,27 @@
+#ifndef _ERROR_H_
+#define _ERROR_H_
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+static unsigned int error_message_count = 0;
+
+static inline void error(int status, int errnum, const char* format, ...)
+{
+	va_list ap;
+	fprintf(stderr, "%s: ", program_invocation_name);
+	va_start(ap, format);
+	vfprintf(stderr, format, ap);
+	va_end(ap);
+	if (errnum)
+		fprintf(stderr, ": %s", strerror(errnum));
+	fprintf(stderr, "\n");
+	error_message_count++;
+	if (status)
+		exit(status);
+}
+
+#endif	/* _ERROR_H_ */
diff -Naur elfutils-0.171.orig/lib/fixedsizehash.h elfutils-0.171/lib/fixedsizehash.h
--- elfutils-0.171.orig/lib/fixedsizehash.h	2018-06-11 10:11:02.222181359 -0700
+++ elfutils-0.171/lib/fixedsizehash.h	2018-06-11 10:11:15.913181114 -0700
@@ -30,7 +30,6 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/cdefs.h>
 
 #include <system.h>
 
diff -Naur elfutils-0.171.orig/libdwfl/dwfl_error.c elfutils-0.171/libdwfl/dwfl_error.c
--- elfutils-0.171.orig/libdwfl/dwfl_error.c	2018-06-11 10:11:02.245181359 -0700
+++ elfutils-0.171/libdwfl/dwfl_error.c	2018-06-11 10:11:18.020181076 -0700
@@ -154,7 +154,16 @@
   switch (error &~ 0xffff)
     {
     case OTHER_ERROR (ERRNO):
+#if defined(__GLIBC__)
       return strerror_r (error & 0xffff, "bad", 0);
+#else
+      {
+        static __thread char buf[128] = "";
+        if (strerror_r (error & 0xffff, buf, sizeof(buf)) == 0)
+          return buf;
+      }
+      return "strerror_r() failed";
+#endif
     case OTHER_ERROR (LIBELF):
       return elf_errmsg (error & 0xffff);
     case OTHER_ERROR (LIBDW):
diff -Naur elfutils-0.171.orig/libelf/elf.h elfutils-0.171/libelf/elf.h
--- elfutils-0.171.orig/libelf/elf.h	2018-06-11 10:11:02.225181359 -0700
+++ elfutils-0.171/libelf/elf.h	2018-06-11 10:11:19.036181058 -0700
@@ -21,7 +21,9 @@
 
 #include <features.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /* Standard ELF types.  */
 
@@ -3912,6 +3914,8 @@
 #define R_METAG_TLS_LE_HI16	60
 #define R_METAG_TLS_LE_LO16	61
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* elf.h */
