diff -Naur elfutils-0.168.orig/src/arlib.h elfutils-0.168/src/arlib.h
--- elfutils-0.168.orig/src/arlib.h	2016-12-28 10:56:21.570727621 -0800
+++ elfutils-0.168/src/arlib.h	2016-12-28 11:51:27.643036959 -0800
@@ -29,6 +29,12 @@
 #include <stdint.h>
 #include <sys/types.h>
 
+#if !defined(ALLPERMS)
+# define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO) /* 07777 */
+#endif
+#if !defined(DEFFILEMODE)
+# define DEFFILEMODE (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)/* 0666*/
+#endif
 
 /* State of -D/-U flags.  */
 extern bool arlib_deterministic_output;
diff -Naur elfutils-0.168.orig/src/elfcompress.c elfutils-0.168/src/elfcompress.c
--- elfutils-0.168.orig/src/elfcompress.c	2016-12-28 10:56:21.569727612 -0800
+++ elfutils-0.168/src/elfcompress.c	2016-12-28 11:53:53.587463560 -0800
@@ -37,6 +37,13 @@
 #include "libeu.h"
 #include "system.h"
 
+#if !defined(ALLPERMS)
+# define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO) /* 07777 */
+#endif
+#if !defined(FNM_EXTMATCH)
+# define FNM_EXTMATCH ( 1 << 5)
+#endif
+
 /* Name and version of program.  */
 ARGP_PROGRAM_VERSION_HOOK_DEF = print_version;
 
diff -Naur elfutils-0.168.orig/src/readelf.c elfutils-0.168/src/readelf.c
--- elfutils-0.168.orig/src/readelf.c	2016-12-28 10:56:21.572727640 -0800
+++ elfutils-0.168/src/readelf.c	2016-12-28 11:45:38.460680025 -0800
@@ -4490,10 +4490,11 @@
   return base;
 }
 
+static const char *listptr_name;
+
 static int
-compare_listptr (const void *a, const void *b, void *arg)
+compare_listptr (const void *a, const void *b)
 {
-  const char *name = arg;
   struct listptr *p1 = (void *) a;
   struct listptr *p2 = (void *) b;
 
@@ -4509,21 +4510,21 @@
 	  p1->warned = p2->warned = true;
 	  error (0, 0,
 		 gettext ("%s %#" PRIx64 " used with different address sizes"),
-		 name, (uint64_t) p1->offset);
+		 listptr_name, (uint64_t) p1->offset);
 	}
       if (p1->dwarf64 != p2->dwarf64)
 	{
 	  p1->warned = p2->warned = true;
 	  error (0, 0,
 		 gettext ("%s %#" PRIx64 " used with different offset sizes"),
-		 name, (uint64_t) p1->offset);
+		 listptr_name, (uint64_t) p1->offset);
 	}
       if (listptr_base (p1) != listptr_base (p2))
 	{
 	  p1->warned = p2->warned = true;
 	  error (0, 0,
 		 gettext ("%s %#" PRIx64 " used with different base addresses"),
-		 name, (uint64_t) p1->offset);
+		 listptr_name, (uint64_t) p1->offset);
 	}
     }
 
@@ -4589,8 +4590,11 @@
 sort_listptr (struct listptr_table *table, const char *name)
 {
   if (table->n > 0)
-    qsort_r (table->table, table->n, sizeof table->table[0],
-	     &compare_listptr, (void *) name);
+    {
+      listptr_name = name;
+      qsort (table->table, table->n, sizeof table->table[0],
+	    &compare_listptr);
+    }
 }
 
 static bool
diff -Naur elfutils-0.168.orig/src/strip.c elfutils-0.168/src/strip.c
--- elfutils-0.168.orig/src/strip.c	2016-12-28 10:56:21.571727631 -0800
+++ elfutils-0.168/src/strip.c	2016-12-28 11:46:09.680979303 -0800
@@ -45,6 +45,10 @@
 #include <libeu.h>
 #include <system.h>
 
+#if !defined(ACCESSPERMS)
+# define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO) /* 0777 */
+#endif
+
 typedef uint8_t GElf_Byte;
 
 /* Name and version of program.  */
diff -Naur elfutils-0.168.orig/src/unstrip.c elfutils-0.168/src/unstrip.c
--- elfutils-0.168.orig/src/unstrip.c	2016-12-28 10:56:21.571727631 -0800
+++ elfutils-0.168/src/unstrip.c	2016-12-28 11:57:40.821684766 -0800
@@ -56,6 +56,15 @@
 # define _(str) gettext (str)
 #endif
 
+#ifndef strndupa
+#define strndupa(s, n) \
+       ({const char *__in = (s); \
+         size_t __len = strnlen (__in, (n)) + 1; \
+         char *__out = (char *) alloca (__len); \
+         __out[__len-1] = '\0'; \
+         (char *) memcpy (__out, __in, __len-1);})
+#endif
+ 
 /* Name and version of program.  */
 ARGP_PROGRAM_VERSION_HOOK_DEF = print_version;
 
