diff -Naurp xfsdump-3.1.4.orig/common/main.c xfsdump-3.1.4.mod/common/main.c
--- xfsdump-3.1.4.orig/common/main.c	2012-10-09 20:42:33.000000000 +0000
+++ xfsdump-3.1.4.mod/common/main.c	2015-05-10 20:14:36.255773000 +0000
@@ -50,6 +50,10 @@
 #include "content.h"
 #include "inventory.h"
 
+#ifndef RLIM64_INFINITY
+#define RLIM64_INFINITY (~0ULL)
+#endif
+
 #ifdef DUMP
 /* main.c - main for dump
  */
@@ -2220,7 +2224,10 @@ static sig_printmap_t sig_printmap[ ] =
 	{SIGTERM,	"SIGTERM"},
 	{SIGUSR1,	"SIGUSR1"},
 	{SIGUSR2,	"SIGUSR2"},
+	{SIGCHLD,	"SIGCHLD"},
+#ifdef SIGCLD
 	{SIGCLD,	"SIGCLD"},
+#endif
 	{SIGPWR,	"SIGPWR"},
 	{SIGURG,	"SIGURG"},
 	{SIGPOLL,	"SIGPOLL"},
diff -Naurp xfsdump-3.1.4.orig/dump/content.c xfsdump-3.1.4.mod/dump/content.c
--- xfsdump-3.1.4.orig/dump/content.c	2015-05-10 19:45:58.915773000 +0000
+++ xfsdump-3.1.4.mod/dump/content.c	2015-05-10 22:23:10.705773000 +0000
@@ -33,6 +33,7 @@
 
 #ifdef linux
 #include <xfs/xqm.h>
+#include <linux/limits.h>
 #endif
 
 #include <attr/attributes.h>
