diff -Naur isapnptools-1.27.orig/include/isapnp/iopl.h isapnptools-1.27/include/isapnp/iopl.h
--- isapnptools-1.27.orig/include/isapnp/iopl.h	2001-05-15 20:37:05.000000000 +0000
+++ isapnptools-1.27/include/isapnp/iopl.h	2018-07-13 05:26:36.497375033 +0000
@@ -52,15 +52,7 @@
 #include <dos.h>
 #else
 #ifndef _OS2_
-#  if defined __GLIBC__ && __GLIBC__ >= 2
-#    include <sys/io.h>
-#  else
-#    ifdef _AXP_
-#       include <sys/io.h>
-#    else
-#       include <asm/io.h>
-#    endif
-#  endif	      	   /* __GLIBC__ */
+#include <sys/io.h>
 #else
 #include <sys/hw.h>
 #endif /* _OS2_ */
diff -Naur isapnptools-1.27.orig/src/pnp-select.c isapnptools-1.27/src/pnp-select.c
--- isapnptools-1.27.orig/src/pnp-select.c	2001-05-15 20:37:04.000000000 +0000
+++ isapnptools-1.27/src/pnp-select.c	2018-07-13 05:27:00.145229137 +0000
@@ -6,6 +6,8 @@
    Foundation.
 */
 
+#include <stdlib.h>
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
