diff -Naur kBuild-0.1.9998-pre20131130.orig/src/ash/jobs.c kBuild-0.1.9998-pre20131130/src/ash/jobs.c
--- kBuild-0.1.9998-pre20131130.orig/src/ash/jobs.c	2016-11-16 12:43:46.369892134 -0800
+++ kBuild-0.1.9998-pre20131130/src/ash/jobs.c	2016-11-16 13:00:43.891220488 -0800
@@ -46,9 +46,6 @@
 #endif /* not lint */
 
 #include <fcntl.h>
-#ifdef __sun__
-#define sys_siglist _sys_siglist
-#endif
 #include <signal.h>
 #include <errno.h>
 #include <unistd.h>
@@ -463,8 +460,8 @@
 #endif
 				st = WTERMSIG(ps->status);
 			st &= 0x7f;
-			if (st < NSIG && sys_siglist[st])
-				scopyn(sys_siglist[st], s + col, 32);
+			if (st < NSIG && strsignal(st))
+				scopyn(strsignal(st), s + col, 32);
 			else
 				fmtstr(s + col, 16, "Signal %d", st);
 			if (WCOREDUMP(ps->status)) {
diff -Naur kBuild-0.1.9998-pre20131130.orig/src/kash/jobs.c kBuild-0.1.9998-pre20131130/src/kash/jobs.c
--- kBuild-0.1.9998-pre20131130.orig/src/kash/jobs.c	2016-11-16 12:43:46.309891526 -0800
+++ kBuild-0.1.9998-pre20131130/src/kash/jobs.c	2016-11-16 13:00:43.892220498 -0800
@@ -396,8 +396,8 @@
 #endif
 				st = WTERMSIG(ps->status);
 			st &= 0x7f;
-			if (st < NSIG && sys_siglist[st])
-				scopyn(sys_siglist[st], s + col, 32);
+			if (st < NSIG && strsignal(st))
+				scopyn(strsignal(st), s + col, 32);
 			else
 				fmtstr(s + col, 16, "Signal %d", st);
 			if (WCOREDUMP(ps->status)) {
diff -Naur kBuild-0.1.9998-pre20131130.orig/src/kmk/kmkbuiltin/chmod.c kBuild-0.1.9998-pre20131130/src/kmk/kmkbuiltin/chmod.c
--- kBuild-0.1.9998-pre20131130.orig/src/kmk/kmkbuiltin/chmod.c	2016-11-16 12:43:46.360892043 -0800
+++ kBuild-0.1.9998-pre20131130/src/kmk/kmkbuiltin/chmod.c	2016-11-16 13:00:48.929272615 -0800
@@ -66,6 +66,10 @@
 #include "getopt.h"
 #include "kmkbuiltin.h"
 
+#if !defined(ALLPERMS)
+# define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO) /* 07777 */
+#endif
+
 extern void * bsd_setmode(const char *p);
 extern mode_t bsd_getmode(const void *bbox, mode_t omode);
 extern void bsd_strmode(mode_t mode, char *p);
diff -Naur kBuild-0.1.9998-pre20131130.orig/src/kmk/kmkbuiltin/install.c kBuild-0.1.9998-pre20131130/src/kmk/kmkbuiltin/install.c
--- kBuild-0.1.9998-pre20131130.orig/src/kmk/kmkbuiltin/install.c	2016-11-16 12:43:46.362892063 -0800
+++ kBuild-0.1.9998-pre20131130/src/kmk/kmkbuiltin/install.c	2016-11-16 13:00:48.929272615 -0800
@@ -127,6 +127,10 @@
 # define IS_SLASH(ch)   ((ch) == '/')
 #endif
 
+#if !defined(ALLPERMS)
+# define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO) /* 07777 */
+#endif
+
 static gid_t gid;
 static uid_t uid;
 static int dobackup, docompare, dodir, dopreserve, dostrip, nommap, safecopy, verbose, mode_given;
