diff -Naur kBuild-0.1.9998-pre20131130.orig/src/kObjCache/kObjCache.c kBuild-0.1.9998-pre20131130/src/kObjCache/kObjCache.c
--- kBuild-0.1.9998-pre20131130.orig/src/kObjCache/kObjCache.c	2016-11-16 12:43:46.303891466 -0800
+++ kBuild-0.1.9998-pre20131130/src/kObjCache/kObjCache.c	2016-11-16 12:48:21.106674374 -0800
@@ -38,6 +38,7 @@
 #include <errno.h>
 #include <assert.h>
 #include <sys/stat.h>
+#include <sys/file.h>
 #include <fcntl.h>
 #include <limits.h>
 #include <ctype.h>
diff -Naur kBuild-0.1.9998-pre20131130.orig/src/kmk/kmkbuiltin/ftsfake.h kBuild-0.1.9998-pre20131130/src/kmk/kmkbuiltin/ftsfake.h
--- kBuild-0.1.9998-pre20131130.orig/src/kmk/kmkbuiltin/ftsfake.h	2016-11-16 12:43:46.362892063 -0800
+++ kBuild-0.1.9998-pre20131130/src/kmk/kmkbuiltin/ftsfake.h	2016-11-16 12:48:39.365859284 -0800
@@ -34,6 +34,8 @@
 #ifndef	_FTS_H_
 #define	_FTS_H_
 
+#include <sys/types.h>
+
 #ifdef _MSC_VER
 # include "kmkbuiltin/mscfakes.h"
 #endif
