Index: iperf-3.1.3/src/iperf.h
===================================================================
--- iperf-3.1.3.orig/src/iperf.h
+++ iperf-3.1.3/src/iperf.h
@@ -36,7 +36,13 @@
 #endif
 #include <sys/select.h>
 #include <sys/socket.h>
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
 
 #if defined(HAVE_CPUSET_SETAFFINITY)
 #include <sys/param.h>
@@ -66,7 +72,7 @@ struct iperf_interval_results
     int       cnt_error;
 
     int omitted;
-#if (defined(linux) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+#if (defined(__GLIBC__) || defined(linux) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
 	defined(TCP_INFO)
     struct tcp_info tcpInfo; /* getsockopt(TCP_INFO) for Linux, {Free,Net}BSD */
 #else
Index: iperf-3.1.3/src/iperf_api.c
===================================================================
--- iperf-3.1.3.orig/src/iperf_api.c
+++ iperf-3.1.3/src/iperf_api.c
@@ -47,7 +47,15 @@
 #ifdef HAVE_STDINT_H
 #include <stdint.h>
 #endif
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
+
+
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <sys/mman.h>
Index: iperf-3.1.3/src/iperf_server_api.c
===================================================================
--- iperf-3.1.3.orig/src/iperf_server_api.c
+++ iperf-3.1.3/src/iperf_server_api.c
@@ -44,7 +44,14 @@
 #ifdef HAVE_STDINT_H
 #include <stdint.h>
 #endif
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
+
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <sched.h>
Index: iperf-3.1.3/src/iperf_tcp.c
===================================================================
--- iperf-3.1.3.orig/src/iperf_tcp.c
+++ iperf-3.1.3/src/iperf_tcp.c
@@ -35,7 +35,14 @@
 #include <sys/types.h>
 #include <netinet/in.h>
 #include <netdb.h>
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
+
 #include <sys/time.h>
 #include <sys/select.h>
 
Index: iperf-3.1.3/src/iperf_sctp.c
===================================================================
--- iperf-3.1.3.orig/src/iperf_sctp.c
+++ iperf-3.1.3/src/iperf_sctp.c
@@ -35,7 +35,14 @@
 #include <sys/types.h>
 #include <netinet/in.h>
 #include <netdb.h>
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
+
 #include <sys/time.h>
 #include <sys/select.h>
 
Index: iperf-3.1.3/src/tcp_info.c
===================================================================
--- iperf-3.1.3.orig/src/tcp_info.c
+++ iperf-3.1.3/src/tcp_info.c
@@ -48,7 +48,14 @@
 #include <sys/param.h>
 #include <sys/types.h>
 #include <sys/socket.h>
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
+
 #include <string.h>
 #include <netinet/in.h>
 #include <errno.h>
Index: iperf-3.1.3/src/units.c
===================================================================
--- iperf-3.1.3.orig/src/units.c
+++ iperf-3.1.3/src/units.c
@@ -60,7 +60,13 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <sys/time.h>
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
 
 
 #include "iperf.h"
Index: iperf-3.1.3/src/main.c
===================================================================
--- iperf-3.1.3.orig/src/main.c
+++ iperf-3.1.3/src/main.c
@@ -44,7 +44,13 @@
 #ifdef HAVE_STDINT_H
 #include <stdint.h>
 #endif
+
+#if (defined(__GLIBC__) || defined(__FreeBSD__) || defined(__NetBSD__)) && \
+	defined(TCP_INFO)
 #include <netinet/tcp.h>
+#elif defined(linux)
+#include <linux/tcp.h>
+#endif
 
 #include "iperf.h"
 #include "iperf_api.h"
Index: iperf-3.1.3/src/timer.h
===================================================================
--- iperf-3.1.3.orig/src/timer.h
+++ iperf-3.1.3/src/timer.h
@@ -32,6 +32,8 @@
 
 #include <sys/time.h>
 
+#include <stdint.h>
+
 /* TimerClientData is an opaque value that tags along with a timer.  The
 ** client can use it for whatever, and it gets passed to the callback when
 ** the timer triggers.
Index: iperf-3.1.3/src/t_uuid.c
===================================================================
--- iperf-3.1.3.orig/src/t_uuid.c
+++ iperf-3.1.3/src/t_uuid.c
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/time.h>
+#include <stdint.h>
 
 #include "iperf_util.h"
 
