diff -Naur bubblewrap-0.3.1.orig/bubblewrap.c bubblewrap-0.3.1/bubblewrap.c
--- bubblewrap-0.3.1.orig/bubblewrap.c	2018-09-26 12:55:17.000000000 -0000
+++ bubblewrap-0.3.1/bubblewrap.c	2019-05-11 14:02:55.475083321 -0000
@@ -37,6 +37,7 @@
 #include "utils.h"
 #include "network.h"
 #include "bind-mount.h"
+#include "missing.h"
 
 #ifndef CLONE_NEWCGROUP
 #define CLONE_NEWCGROUP 0x02000000 /* New cgroup namespace */
diff -Naur bubblewrap-0.3.1.orig/missing.h bubblewrap-0.3.1/missing.h
--- bubblewrap-0.3.1.orig/missing.h	1970-01-01 00:00:00.000000000 -0000
+++ bubblewrap-0.3.1/missing.h	2019-05-11 14:00:21.001061569 -0000
@@ -0,0 +1,6 @@
+# define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
