taken from sabotage linux
--- wine-1.5.19/dlls/ntdll/directory.c	2012-12-07 19:17:32.000000000 +0000
+++ wine-1.5.19.patched/dlls/ntdll/directory.c	2012-12-19 04:47:29.148000002 +0000
@@ -119,7 +119,7 @@
     char           d_name[256];
 } KERNEL_DIRENT64;
 
-static inline int getdents64( int fd, char *de, unsigned int size )
+static inline int BUG_getdents64( int fd, char *de, unsigned int size )
 {
     return syscall( SYS_getdents64, fd, de, size );
 }
