diff -Naur portage-2.2.26.orig/pym/portage/util/locale.py portage-2.2.26/pym/portage/util/locale.py
--- portage-2.2.26.orig/pym/portage/util/locale.py	2015-11-24 16:38:22.000000000 -0500
+++ portage-2.2.26/pym/portage/util/locale.py	2016-01-20 20:08:08.731746146 -0500
@@ -13,10 +13,10 @@
 import os
 import textwrap
 import traceback
+from ctypes import CDLL
 
 import portage
 from portage.util import writemsg_level
-from portage.util._ctypes import find_library, LoadLibrary
 
 
 locale_categories = (
@@ -35,12 +35,7 @@
 	The inner locale check function.
 	"""
 
-	libc_fn = find_library("c")
-	if libc_fn is None:
-		return None
-	libc = LoadLibrary(libc_fn)
-	if libc is None:
-		return None
+	libc = CDLL("libc.so.6")
 
 	lc = list(range(ord('a'), ord('z')+1))
 	uc = list(range(ord('A'), ord('Z')+1))
