diff -Naur thin-provisioning-tools-0.3.2.orig/caching/cache_check.cc thin-provisioning-tools-0.3.2/caching/cache_check.cc
--- thin-provisioning-tools-0.3.2.orig/caching/cache_check.cc	2014-04-02 08:43:16.000000000 -0400
+++ thin-provisioning-tools-0.3.2/caching/cache_check.cc	2015-07-19 22:42:11.606851760 -0400
@@ -201,10 +201,10 @@
 		int r = ::stat(path.c_str(), &info);
 		if (r) {
 			ostringstream msg;
-			char buffer[128], *ptr;
+			char buffer[128];
 
-			ptr = ::strerror_r(errno, buffer, sizeof(buffer));
-			msg << path << ": " << ptr;
+			::strerror_r(errno, buffer, sizeof(buffer));
+			msg << path << ": " << buffer;
 			throw runtime_error(msg.str());
 		}
 
diff -Naur thin-provisioning-tools-0.3.2.orig/persistent-data/block.tcc thin-provisioning-tools-0.3.2/persistent-data/block.tcc
--- thin-provisioning-tools-0.3.2.orig/persistent-data/block.tcc	2014-04-02 08:43:16.000000000 -0400
+++ thin-provisioning-tools-0.3.2/persistent-data/block.tcc	2015-07-19 22:39:10.567860436 -0400
@@ -45,10 +45,10 @@
 	// to exception.h
 	void syscall_failed(char const *call) {
 		char buffer[128];
-		char *msg = strerror_r(errno, buffer, sizeof(buffer));
+		strerror_r(errno, buffer, sizeof(buffer));
 
 		ostringstream out;
-		out << "syscall '" << call << "' failed: " << msg;
+		out << "syscall '" << call << "' failed: " << buffer;
 		throw runtime_error(out.str());
 	}
 
