diff -Naur thin-provisioning-tools-0.7.0.orig/block-cache/io_engine.h thin-provisioning-tools-0.7.0/block-cache/io_engine.h
--- thin-provisioning-tools-0.7.0.orig/block-cache/io_engine.h	2017-03-27 06:00:54.000000000 -0700
+++ thin-provisioning-tools-0.7.0/block-cache/io_engine.h	2017-08-16 10:21:17.536936717 -0700
@@ -10,6 +10,10 @@
 #include <libaio.h>
 #include <vector>
 
+#ifdef PAGE_SIZE
+#undef PAGE_SIZE
+#endif
+
 //----------------------------------------------------------------
 
 namespace bcache {
