diff -ENwbur gcc-4.8.3.orig/libsanitizer/tsan/tsan_platform_linux.cc gcc-4.8.3/libsanitizer/tsan/tsan_platform_linux.cc
--- gcc-4.8.3.orig/libsanitizer/tsan/tsan_platform_linux.cc	2014-08-31 12:50:18.381689906 +0000
+++ gcc-4.8.3/libsanitizer/tsan/tsan_platform_linux.cc	2014-08-31 13:12:19.838446253 +0000
@@ -292,7 +292,7 @@
 #ifndef TSAN_GO
 int ExtractResolvFDs(void *state, int *fds, int nfd) {
   int cnt = 0;
-  __res_state *statp = (__res_state*)state;
+  res_state statp = (res_state)state;
   for (int i = 0; i < MAXNS && cnt < nfd; i++) {
     if (statp->_u._ext.nsaddrs[i] && statp->_u._ext.nssocks[i] != -1)
       fds[cnt++] = statp->_u._ext.nssocks[i];
