--- /var/tmp/portage/sys-devel/gcc-4.8.3-r99/work/gcc-4.8.3/boehm-gc/os_dep.c	2013-03-06 15:08:58.000000000 +0000
+++ gcc-4.8.3.new/boehm-gc/os_dep.c	2014-10-24 18:25:39.963379871 +0000
@@ -41,7 +41,7 @@
 #     else /* not 2 <= __GLIBC__ */
         /* libc5 doesn't have <sigcontext.h>: go directly with the kernel   */
         /* one.  Check LINUX_VERSION_CODE to see which we should reference. */
-#       include <asm/sigcontext.h>
+//#       include <asm/sigcontext.h>
 #     endif /* 2 <= __GLIBC__ */
 #   endif
 # endif
