diff -Naur ell-0.6.orig/ell/dhcp-transport.c ell-0.6/ell/dhcp-transport.c
--- ell-0.6.orig/ell/dhcp-transport.c	2018-06-26 14:29:02.357862603 -0700
+++ ell-0.6/ell/dhcp-transport.c	2018-06-26 14:29:09.657862472 -0700
@@ -184,12 +184,12 @@
 	ip->saddr = L_CPU_TO_BE32(saddr);
 	ip->daddr = L_CPU_TO_BE32(daddr);
 
-	udp->source = L_CPU_TO_BE16(sport);
-	udp->dest = L_CPU_TO_BE16(dport);
+	udp->uh_sport = L_CPU_TO_BE16(sport);
+	udp->uh_dport = L_CPU_TO_BE16(dport);
 
-	udp->len = L_CPU_TO_BE16(len + sizeof(*udp));
+	udp->uh_ulen = L_CPU_TO_BE16(len + sizeof(*udp));
 
-	ip->check = udp->len;
+	ip->check = udp->uh_ulen;
 
 	iov[0].iov_base = &ip->ttl;
 	iov[0].iov_len = sizeof(*ip) - 8;
@@ -197,7 +197,7 @@
 	iov[1].iov_len = sizeof(*udp);
 	iov[2].iov_base = (void *) data;
 	iov[2].iov_len = len;
-	udp->check = _dhcp_checksumv(iov, 3);
+	udp->uh_sum = _dhcp_checksumv(iov, 3);
 
 	ip->ttl = IPDEFTTL;
 	ip->check = 0;
diff -Naur ell-0.6.orig/ell/dhcp.c ell-0.6/ell/dhcp.c
--- ell-0.6.orig/ell/dhcp.c	2018-06-26 14:29:02.360862602 -0700
+++ ell-0.6/ell/dhcp.c	2018-06-26 14:29:15.918862360 -0700
@@ -27,7 +27,6 @@
 #include <netinet/ip.h>
 #include <net/ethernet.h>
 #include <linux/types.h>
-#include <linux/if_arp.h>
 #include <errno.h>
 #include <time.h>
 
