diff -Naur weston-1.11.0.orig/shared/xalloc.h weston-1.11.0/shared/xalloc.h
--- weston-1.11.0.orig/shared/xalloc.h	2017-03-05 16:35:25.816193000 -0800
+++ weston-1.11.0/shared/xalloc.h	2017-03-05 16:38:16.043363715 -0800
@@ -30,6 +30,7 @@
 extern "C" {
 #endif
 
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
diff -Naur weston-1.11.0.orig/src/error.h weston-1.11.0/src/error.h
--- weston-1.11.0.orig/src/error.h	1969-12-31 16:00:00.000000000 -0800
+++ weston-1.11.0/src/error.h	2017-03-05 16:37:51.441046969 -0800
@@ -0,0 +1,27 @@
+#ifndef _ERROR_H_
+#define _ERROR_H_
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+static unsigned int error_message_count = 0;
+
+static inline void error(int status, int errnum, const char* format, ...)
+{
+	va_list ap;
+	fprintf(stderr, "%s: ", program_invocation_name);
+	va_start(ap, format);
+	vfprintf(stderr, format, ap);
+	va_end(ap);
+	if (errnum)
+		fprintf(stderr, ": %s", strerror(errnum));
+	fprintf(stderr, "\n");
+	error_message_count++;
+	if (status)
+		exit(status);
+}
+
+#endif	/* _ERROR_H_ */
