diff -Naur ruby-2.0.0-p247.orig/missing/isinf.c ruby-2.0.0-p247/missing/isinf.c
--- ruby-2.0.0-p247.orig/missing/isinf.c	2011-05-15 07:55:52.000000000 -0400
+++ ruby-2.0.0-p247/missing/isinf.c	2013-09-09 15:28:47.750452951 -0400
@@ -52,6 +52,8 @@
 static double one (void) { return 1.0; }
 static double inf (void) { return one() / zero(); }
 
+#undef isinf
+
 int
 isinf(double n)
 {
diff -Naur ruby-2.0.0-p247.orig/missing/isnan.c ruby-2.0.0-p247/missing/isnan.c
--- ruby-2.0.0-p247.orig/missing/isnan.c	2010-07-28 04:12:01.000000000 -0400
+++ ruby-2.0.0-p247/missing/isnan.c	2013-09-09 15:28:32.791451837 -0400
@@ -4,6 +4,8 @@
 
 static int double_ne(double n1, double n2);
 
+#undef isnan
+
 int
 isnan(double n)
 {
