From 63ba9622d49270a1f9795da9aab8f9934c6e088b Mon Sep 17 00:00:00 2001
From: Jory Pratt <anarchy@gentoo.org>
Date: Thu, 7 May 2020 18:50:57 -0500
Subject: [PATCH] tevent.h needs sys/types.h for pid_t

Signed-off-by: Jory Pratt <anarchy@gentoo.org>
---
 tevent.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tevent.h b/tevent.h
index 3c3e3cc..011e1ad 100644
--- a/tevent.h
+++ b/tevent.h
@@ -31,6 +31,7 @@
 #include <stdint.h>
 #include <talloc.h>
 #include <sys/time.h>
+#include <sys/types.h>
 #include <stdbool.h>
 
 struct tevent_context;
-- 
2.26.2

