diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/man/Makefile.am libbsd-0.8.2/man/Makefile.am
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/man/Makefile.am	2015-11-30 22:48:50.000000000 +0000
+++ libbsd-0.8.2/man/Makefile.am	2016-04-05 00:23:52.848000000 +0000
@@ -29,7 +29,6 @@
 	flopen.3 \
 	fmtcheck.3 \
 	fparseln.3 \
-	funopen.3bsd \
 	getbsize.3 \
 	getmode.3 \
 	getpeereid.3 \
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/Makefile.am libbsd-0.8.2/src/Makefile.am
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/Makefile.am	2015-12-14 02:34:40.000000000 +0000
+++ libbsd-0.8.2/src/Makefile.am	2016-04-05 00:14:59.360000000 +0000
@@ -74,7 +74,6 @@
 	fmtcheck.c \
 	fparseln.c \
 	fpurge.c \
-	funopen.c \
 	getbsize.c \
 	getpeereid.c \
 	hash/md5.c \
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/test/Makefile.am libbsd-0.8.2/test/Makefile.am
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/test/Makefile.am	2015-12-14 02:34:40.000000000 +0000
+++ libbsd-0.8.2/test/Makefile.am	2016-04-05 00:24:03.592000000 +0000
@@ -36,7 +36,6 @@
 	endian \
 	humanize \
 	fgetln \
-	funopen \
 	fparseln \
 	fpurge \
 	proctitle-init \
