diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/sys/cdefs.h libbsd-0.8.2/include/bsd/sys/cdefs.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/sys/cdefs.h	2015-09-23 05:59:27.000000000 +0000
+++ libbsd-0.8.2/include/bsd/sys/cdefs.h	2016-04-04 22:04:30.264000000 +0000
@@ -24,11 +24,13 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+/*
 #ifdef LIBBSD_OVERLAY
 #include_next <sys/cdefs.h>
 #else
 #include <sys/cdefs.h>
 #endif
+*/
 
 #ifndef LIBBSD_SYS_CDEFS_H
 #define LIBBSD_SYS_CDEFS_H
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/flopen.c libbsd-0.8.2/src/flopen.c
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/flopen.c	2011-07-05 16:56:44.000000000 +0000
+++ libbsd-0.8.2/src/flopen.c	2016-04-04 23:46:26.120000000 +0000
@@ -35,6 +35,7 @@
 #include <stdarg.h>
 #include <unistd.h>
 #include <libutil.h>
+#include <fcntl.h>
 
 int
 flopen(const char *path, int flags, ...)
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/nlist.c libbsd-0.8.2/src/nlist.c
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/nlist.c	2015-09-23 05:59:34.000000000 +0000
+++ libbsd-0.8.2/src/nlist.c	2016-04-05 00:22:48.056000000 +0000
@@ -39,10 +39,11 @@
 #include <arpa/inet.h>
 
 #include <errno.h>
-#include <a.out.h>
+#include <linux/a.out.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <fcntl.h>
 
 #if !defined(__NO_A_OUT_SUPPORT)
 #define _NLIST_DO_AOUT
