diff -Naurw connman-1.29.orig/src/ntp.c connman-1.29/src/ntp.c
--- connman-1.29.orig/src/ntp.c	2014-12-13 01:23:18.000000000 +0000
+++ connman-1.29/src/ntp.c	2015-08-12 16:17:19.636686053 +0000
@@ -180,7 +180,7 @@
 	msg.xmttime.fraction = htonl(transmit_timeval.tv_usec * 1000);
 
 	len = sendto(fd, &msg, sizeof(msg), MSG_DONTWAIT,
-						&addr, sizeof(addr));
+						(struct sockaddr *) &addr, sizeof(addr));
 	if (len < 0) {
 		connman_error("Time request for server %s failed (%d/%s)",
 			server, errno, strerror(errno));
