diff -Naur kbd-1.15.3.orig/src/dumpkeys.c kbd-1.15.3/src/dumpkeys.c
--- kbd-1.15.3.orig/src/dumpkeys.c	2011-05-07 20:33:42.000000000 +0000
+++ kbd-1.15.3/src/dumpkeys.c	2013-07-06 02:20:41.660131968 +0000
@@ -12,6 +12,7 @@
 #include <linux/kd.h>
 #include <linux/keyboard.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <string.h>
 #include <errno.h>
 #include "ksyms.h"
diff -Naur kbd-1.15.3.orig/src/kbdinfo.c kbd-1.15.3/src/kbdinfo.c
--- kbd-1.15.3.orig/src/kbdinfo.c	2011-05-07 20:33:42.000000000 +0000
+++ kbd-1.15.3/src/kbdinfo.c	2013-07-06 02:29:12.057160508 +0000
@@ -1,6 +1,5 @@
 #include <stdio.h>
 #include <errno.h>
-#include <error.h>
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <linux/kd.h>
@@ -9,6 +8,10 @@
 #include "nls.h"
 #include "version.h"
 
+#define error(e, n, s)	\
+	perror(s);	\
+	exit(e);	\
+
 static const char *action = NULL;
 static const char *value  = NULL;
 
diff -Naur kbd-1.15.3.orig/src/kbdrate.c kbd-1.15.3/src/kbdrate.c
--- kbd-1.15.3.orig/src/kbdrate.c	2011-05-07 20:33:42.000000000 +0000
+++ kbd-1.15.3/src/kbdrate.c	2013-07-06 02:21:53.121135964 +0000
@@ -74,6 +74,7 @@
 #include <errno.h>
 #include <sys/file.h>
 #include <sys/ioctl.h>
+#include <fcntl.h>
 
 #ifdef __sparc__
 #include <asm/param.h>
diff -Naur kbd-1.15.3.orig/src/loadkeys.c kbd-1.15.3/src/loadkeys.c
--- kbd-1.15.3.orig/src/loadkeys.c	2011-05-07 20:33:42.000000000 +0000
+++ kbd-1.15.3/src/loadkeys.c	2013-07-06 02:21:02.148133114 +0000
@@ -79,6 +79,7 @@
 #include <ctype.h>
 #include <sys/param.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <linux/kd.h>
 #include <linux/keyboard.h>
 #include <unistd.h>
diff -Naur kbd-1.15.3.orig/src/openvt.c kbd-1.15.3/src/openvt.c
--- kbd-1.15.3.orig/src/openvt.c	2011-05-07 20:33:42.000000000 +0000
+++ kbd-1.15.3/src/openvt.c	2013-07-06 02:22:11.080136968 +0000
@@ -10,6 +10,7 @@
 #include <sys/vt.h>
 #include <sys/wait.h>
 #include <sys/file.h>
+#include <fcntl.h>
 
 #include "version.h"
 #include "xmalloc.h"
diff -Naur kbd-1.15.3.orig/src/setvtrgb.c kbd-1.15.3/src/setvtrgb.c
--- kbd-1.15.3.orig/src/setvtrgb.c	2011-05-07 20:33:42.000000000 +0000
+++ kbd-1.15.3/src/setvtrgb.c	2013-07-06 02:34:26.390178084 +0000
@@ -5,12 +5,15 @@
 #include <sys/ioctl.h>
 #include <linux/kd.h>
 #include <errno.h>
-#include <error.h>
 #include "kbd.h"
 #include "getfd.h"
 #include "nls.h"
 #include "version.h"
 
+#define error(e, n, s)  \
+	perror(s);      \
+        exit(e);        \
+
 static unsigned char *cmap;
 
 /* Standard VGA terminal colors, matching those hardcoded in the Linux kernel's
@@ -82,25 +85,21 @@
 				if (c == EOF)
 					error(EXIT_FAILURE, errno, "fscanf");
 
-				error(EXIT_FAILURE, 0, _("Error: %s: Invalid value in field %u in line %u."),
-				      filename, rows + 1, cols + 1);
+				error(EXIT_FAILURE, 0, "setvtrgb");
 			}
 
 			cmap[rows + cols * 3] = (unsigned char) val;
 
 			if (cols < 15 && fgetc(fd) != ',')
-				error(EXIT_FAILURE, 0, _("Error: %s: Insufficient number of fields in line %u."),
-				      filename, rows + 1);
+				error(EXIT_FAILURE, 0, "setvtrgb");
 			cols++;
 		}
 
 		if ((c = fgetc(fd)) == EOF)
-			error(EXIT_FAILURE, 0, _("Error: %s: Line %u has ended unexpectedly.\n"),
-			      filename, rows + 1); 
+			error(EXIT_FAILURE, 0, "setvtrgb");
 
 		if (c != '\n')
-			error(EXIT_FAILURE, 0, _("Error: %s: Line %u is too long.\n"),
-			      filename, rows + 1);
+			error(EXIT_FAILURE, 0, "setvtrgb");
 	}
 }
 
