diff -ur a/llvm-3.5.0.src/projects/compiler-rt/lib/Makefile.mk b/llvm-3.5.0.src/projects/compiler-rt/lib/Makefile.mk
--- a/projects/compiler-rt/lib/Makefile.mk	2014-02-14 10:20:33.000000000 -0100
+++ b/projects/compiler-rt/lib/Makefile.mk	2015-01-07 13:56:00.362944069 -0100
@@ -10,13 +10,13 @@
 SubDirs :=
 
 # Add submodules.
-SubDirs += asan
+#SubDirs += asan
 SubDirs += builtins
-SubDirs += dfsan
+#SubDirs += dfsan
 SubDirs += interception
-SubDirs += lsan
-SubDirs += msan
+#SubDirs += lsan
+#SubDirs += msan
 SubDirs += profile
-SubDirs += sanitizer_common
-SubDirs += tsan
-SubDirs += ubsan
+#SubDirs += sanitizer_common
+#SubDirs += tsan
+#SubDirs += ubsan
diff -ur a/llvm-3.5.0.src/projects/compiler-rt/lib/interception/interception_linux.cc b/llvm-3.5.0.src/projects/compiler-rt/lib/interception/interception_linux.cc
--- a/projects/compiler-rt/lib/interception/interception_linux.cc	2014-02-24 09:37:41.000000000 -0100
+++ b/projects/compiler-rt/lib/interception/interception_linux.cc	2015-01-07 13:56:56.609240893 -0100
@@ -24,12 +24,6 @@
   return real == wrapper;
 }

-#if !defined(__ANDROID__)  // android does not have dlvsym
-void *GetFuncAddrVer(const char *func_name, const char *ver) {
-  return dlvsym(RTLD_NEXT, func_name, ver);
-}
-#endif  // !defined(__ANDROID__)
-
 }  // namespace __interception


diff -ur a/llvm-3.5.0.src/projects/compiler-rt/lib/interception/interception_linux.h b/llvm-3.5.0.src/projects/compiler-rt/lib/interception/interception_linux.h
--- a/projects/compiler-rt/lib/interception/interception_linux.h	2014-02-24 09:37:41.000000000 -0100
+++ b/projects/compiler-rt/lib/interception/interception_linux.h	2015-01-07 13:56:41.617428317 -0100
@@ -34,14 +34,8 @@
       (::__interception::uptr) & (func),                                   \
       (::__interception::uptr) & WRAP(func))

-#if !defined(__ANDROID__)  // android does not have dlvsym
-# define INTERCEPT_FUNCTION_VER_LINUX_OR_FREEBSD(func, symver) \
-     ::__interception::real_##func = (func##_f)(unsigned long) \
-         ::__interception::GetFuncAddrVer(#func, symver)
-#else
 # define INTERCEPT_FUNCTION_VER_LINUX_OR_FREEBSD(func, symver) \
      INTERCEPT_FUNCTION_LINUX_OR_FREEBSD(func)
-#endif  // !defined(__ANDROID__)

 #endif  // INTERCEPTION_LINUX_H
 #endif  // __linux__ || __FreeBSD__
