diff -Naur /var/tmp/portage/net-fs/cifs-utils-6.4/work/cifs-utils-6.4/getcifsacl.c cifs-utils-6.4/getcifsacl.c
--- /var/tmp/portage/net-fs/cifs-utils-6.4/work/cifs-utils-6.4/getcifsacl.c	2014-07-11 15:14:06.000000000 +0000
+++ cifs-utils-6.4/getcifsacl.c	2016-04-05 10:43:29.952000000 +0000
@@ -33,6 +33,8 @@
 #include <stddef.h>
 #include <errno.h>
 #include <limits.h>
+#include <linux/limits.h>
+#include <endian.h>
 #include <ctype.h>
 #include <sys/xattr.h>
 #include "cifsacl.h"
diff -Naur /var/tmp/portage/net-fs/cifs-utils-6.4/work/cifs-utils-6.4/mtab.c cifs-utils-6.4/mtab.c
--- /var/tmp/portage/net-fs/cifs-utils-6.4/work/cifs-utils-6.4/mtab.c	2014-07-11 15:14:06.000000000 +0000
+++ cifs-utils-6.4/mtab.c	2016-04-05 10:02:35.872000000 +0000
@@ -38,6 +38,7 @@
 #include <mntent.h>
 #include <stdlib.h>
 #include <signal.h>
+#include <paths.h>
 #include "mount.h"
 #include "config.h"
 
diff -Naur /var/tmp/portage/net-fs/cifs-utils-6.4/work/cifs-utils-6.4/setcifsacl.c cifs-utils-6.4/setcifsacl.c
--- /var/tmp/portage/net-fs/cifs-utils-6.4/work/cifs-utils-6.4/setcifsacl.c	2014-07-11 15:14:06.000000000 +0000
+++ cifs-utils-6.4/setcifsacl.c	2016-04-05 10:43:45.696000000 +0000
@@ -33,6 +33,8 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <limits.h>
+#include <linux/limits.h>
+#include <endian.h>
 #include <ctype.h>
 #include <sys/xattr.h>
 
