diff -Naur cifs-utils-6.7.orig/getcifsacl.c cifs-utils-6.7/getcifsacl.c
--- cifs-utils-6.7.orig/getcifsacl.c	2017-03-02 16:51:05.000000000 -0000
+++ cifs-utils-6.7/getcifsacl.c	2019-05-11 14:18:04.522455988 -0000
@@ -35,6 +35,9 @@
 #include <limits.h>
 #include <ctype.h>
 #include <sys/xattr.h>
+#include <linux/limits.h>
+#include <endian.h>
+
 #include "cifsacl.h"
 #include "idmap_plugin.h"
 
diff -Naur cifs-utils-6.7.orig/setcifsacl.c cifs-utils-6.7/setcifsacl.c
--- cifs-utils-6.7.orig/setcifsacl.c	2017-03-02 16:51:05.000000000 -0000
+++ cifs-utils-6.7/setcifsacl.c	2019-05-11 14:17:58.285761823 -0000
@@ -35,6 +35,8 @@
 #include <limits.h>
 #include <ctype.h>
 #include <sys/xattr.h>
+#include <linux/limits.h>
+#include <endian.h>
 
 #include "cifsacl.h"
 #include "idmap_plugin.h"
