diff -Naur libselinux-2.7-rc1.orig/src/Makefile libselinux-2.7-rc1/src/Makefile
--- libselinux-2.7-rc1.orig/src/Makefile	2017-06-13 10:16:48.664402029 -0700
+++ libselinux-2.7-rc1/src/Makefile	2017-06-13 10:18:01.124009953 -0700
@@ -144,7 +144,7 @@
 	$(RANLIB) $@
 
 $(LIBSO): $(LOBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ $(PCRE_LDLIBS) -ldl -Wl,$(LD_SONAME_FLAGS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ $(PCRE_LDLIBS) $(FTS_LDLIBS) -ldl -Wl,$(LD_SONAME_FLAGS)
 	ln -sf $@ $(TARGET)
 
 $(LIBPC): $(LIBPC).in ../VERSION
