diff -Naurw caps-0.9.15.orig/basics.h caps-0.9.15/basics.h
--- caps-0.9.15.orig/basics.h	2013-01-11 05:39:06.000000000 -0800
+++ caps-0.9.15/basics.h	2015-12-24 14:20:06.871407126 -0800
@@ -49,6 +49,18 @@
 
 #include "ladspa.h"
 
+#if !defined(__GLIBC__)
+#include <stdint.h>
+#define __int8_t int8_t
+#define __uint8_t uint8_t
+#define __int16_t int16_t
+#define __uint16_t uint16_t
+#define __int32_t int32_t
+#define __uint32_t uint32_t
+#define __int64_t int64_t
+#define __uint64_t uint64_t
+#endif
+
 typedef __int8_t			int8;
 typedef __uint8_t			uint8;
 typedef __int16_t			int16;
