diff -Naur guvcview-src-2.0.4.orig/guvcview/guvcview.c guvcview-src-2.0.4/guvcview/guvcview.c
--- guvcview-src-2.0.4.orig/guvcview/guvcview.c	2016-04-25 08:58:04.000000000 -0700
+++ guvcview-src-2.0.4/guvcview/guvcview.c	2016-08-29 21:38:12.423067646 -0700
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <signal.h>
 #include <unistd.h>
+#include <locale.h>
 #include <sys/syscall.h>
 #include <sys/stat.h>
 #include <sys/resource.h>
diff -Naur guvcview-src-2.0.4.orig/gview_audio/gviewaudio.h guvcview-src-2.0.4/gview_audio/gviewaudio.h
--- guvcview-src-2.0.4.orig/gview_audio/gviewaudio.h	2016-04-25 08:58:03.000000000 -0700
+++ guvcview-src-2.0.4/gview_audio/gviewaudio.h	2016-08-29 21:38:12.424067647 -0700
@@ -39,7 +39,9 @@
 #include <pthread.h>
 
 /*make sure we support c++*/
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /*Audio API*/
 #define AUDIO_NONE          (0)
@@ -375,6 +377,8 @@
  */
 void audio_close(audio_context_t *audio_ctx);
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif
diff -Naur guvcview-src-2.0.4.orig/gview_encoder/gviewencoder.h guvcview-src-2.0.4/gview_encoder/gviewencoder.h
--- guvcview-src-2.0.4.orig/gview_encoder/gviewencoder.h	2016-04-25 08:58:06.000000000 -0700
+++ guvcview-src-2.0.4/gview_encoder/gviewencoder.h	2016-08-29 21:38:12.424067647 -0700
@@ -38,7 +38,9 @@
 #include <sys/types.h>
 
 /*make sure we support c++*/
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /*encoder modes*/
 #define ENCODER_MODE_NONE   (0)
@@ -717,6 +719,8 @@
  */
 int encoder_disk_supervisor(int treshold, const char *path);
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif
diff -Naur guvcview-src-2.0.4.orig/gview_render/gviewrender.h guvcview-src-2.0.4/gview_render/gviewrender.h
--- guvcview-src-2.0.4.orig/gview_render/gviewrender.h	2016-04-25 08:58:06.000000000 -0700
+++ guvcview-src-2.0.4/gview_render/gviewrender.h	2016-08-29 21:38:12.425067649 -0700
@@ -38,7 +38,9 @@
 #include <sys/types.h>
 
 /*make sure we support c++*/
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 #define RENDER_NONE     (0)
 #define RENDER_SDL      (1)
@@ -313,6 +315,8 @@
  */
 void render_close();
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif
diff -Naur guvcview-src-2.0.4.orig/gview_v4l2core/gviewv4l2core.h guvcview-src-2.0.4/gview_v4l2core/gviewv4l2core.h
--- guvcview-src-2.0.4.orig/gview_v4l2core/gviewv4l2core.h	2016-04-25 08:58:06.000000000 -0700
+++ guvcview-src-2.0.4/gview_v4l2core/gviewv4l2core.h	2016-08-29 21:38:12.425067649 -0700
@@ -33,7 +33,9 @@
 #include <sys/types.h>
 
 /*make sure we support c++*/
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /*
  * LOGITECH Dynamic controls defs
@@ -1423,7 +1425,9 @@
  */
 uint64_t v4l2core_time_get_timestamp();
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif
 
