http://sourceforge.net/p/hunspell/bugs/264/

diff --git a/src/tools/munch.c b/src/tools/munch.c
index 52e7c51..dd42fed 100644
--- a/src/tools/munch.c
+++ b/src/tools/munch.c
@@ -10,8 +10,10 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <limits.h>
-#if defined(__linux__) && !defined(__ANDROID__)
+#if HAVE_ERROR_H
 #include <error.h>
+#endif
+#if defined(__linux__) && !defined(__ANDROID__)
 #include <errno.h>
 #include <sys/mman.h>
 #endif
diff --git a/src/tools/unmunch.c b/src/tools/unmunch.c
index c1d07a8..53915f5 100644
--- a/src/tools/unmunch.c
+++ b/src/tools/unmunch.c
@@ -11,8 +11,10 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
-#if defined(__linux__) && !defined(__ANDROID__)
+#if HAVE_ERROR_H
 #include <error.h>
+#endif
+#if defined(__linux__) && !defined(__ANDROID__)
 #include <errno.h>
 #include <sys/mman.h>
 #endif
-- 
2.0.5

