diff -Naur qtbase-opensource-src-5.9.3.orig/src/corelib/codecs/qiconvcodec.cpp qtbase-opensource-src-5.9.3/src/corelib/codecs/qiconvcodec.cpp
--- qtbase-opensource-src-5.9.3.orig/src/corelib/codecs/qiconvcodec.cpp	2017-11-15 21:15:28.000000000 -0800
+++ qtbase-opensource-src-5.9.3/src/corelib/codecs/qiconvcodec.cpp	2017-11-28 22:24:13.734884247 -0800
@@ -63,7 +63,7 @@
 #elif defined(Q_OS_AIX)
 #  define NO_BOM
 #  define UTF16 "UCS-2"
-#elif defined(Q_OS_FREEBSD)
+#elif defined(Q_OS_FREEBSD) || (defined(Q_OS_LINUX) && !defined(__GLIBC__))
 #  define NO_BOM
 #  if Q_BYTE_ORDER == Q_BIG_ENDIAN
 #    define UTF16 "UTF-16BE"
