diff -ru a/mkspecs/linux-cxx/qplatformdefs.h b/mkspecs/linux-cxx/qplatformdefs.h
--- a/mkspecs/linux-cxx/qplatformdefs.h	2015-05-07 14:14:41.000000000 +0000
+++ b/mkspecs/linux-cxx/qplatformdefs.h	2015-07-04 21:44:04.370713742 +0000
@@ -87,10 +87,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
diff -ru a/mkspecs/linux-ecc-64/qplatformdefs.h b/mkspecs/linux-ecc-64/qplatformdefs.h
--- a/mkspecs/linux-ecc-64/qplatformdefs.h	2015-05-07 14:14:42.000000000 +0000
+++ b/mkspecs/linux-ecc-64/qplatformdefs.h	2015-07-04 21:44:41.610717930 +0000
@@ -87,10 +87,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
diff -ru a/mkspecs/linux-g++/qplatformdefs.h b/mkspecs/linux-g++/qplatformdefs.h
--- a/mkspecs/linux-g++/qplatformdefs.h	2015-05-07 14:14:41.000000000 +0000
+++ b/mkspecs/linux-g++/qplatformdefs.h	2015-07-04 21:45:09.554721073 +0000
@@ -86,10 +86,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
diff -ru a/mkspecs/linux-kcc/qplatformdefs.h b/mkspecs/linux-kcc/qplatformdefs.h
--- a/mkspecs/linux-kcc/qplatformdefs.h	2015-05-07 14:14:41.000000000 +0000
+++ b/mkspecs/linux-kcc/qplatformdefs.h	2015-07-04 21:46:42.865731566 +0000
@@ -90,10 +90,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
diff -ru a/mkspecs/linux-llvm/qplatformdefs.h b/mkspecs/linux-llvm/qplatformdefs.h
--- a/mkspecs/linux-llvm/qplatformdefs.h	2015-05-07 14:14:42.000000000 +0000
+++ b/mkspecs/linux-llvm/qplatformdefs.h	2015-07-04 21:47:02.930733823 +0000
@@ -87,10 +87,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
diff -ru a/mkspecs/linux-lsb-g++/qplatformdefs.h b/mkspecs/linux-lsb-g++/qplatformdefs.h
--- a/mkspecs/linux-lsb-g++/qplatformdefs.h	2015-05-07 14:14:42.000000000 +0000
+++ b/mkspecs/linux-lsb-g++/qplatformdefs.h	2015-07-04 21:47:25.465736357 +0000
@@ -94,10 +94,10 @@
 
 #define QT_OPEN_LARGEFILE       0
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #ifndef SIOCGIFBRDADDR
diff -ru a/mkspecs/linux-pgcc/qplatformdefs.h b/mkspecs/linux-pgcc/qplatformdefs.h
--- a/mkspecs/linux-pgcc/qplatformdefs.h	2015-05-07 14:14:42.000000000 +0000
+++ b/mkspecs/linux-pgcc/qplatformdefs.h	2015-07-04 21:47:53.505739511 +0000
@@ -87,10 +87,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
