diff -Naur libbsd-0.8.3.orig/src/funopen.c libbsd-0.8.3/src/funopen.c
--- libbsd-0.8.3.orig/src/funopen.c	2016-08-10 13:54:09.064111778 -0700
+++ libbsd-0.8.3/src/funopen.c	2016-08-10 14:02:55.416520029 -0700
@@ -137,6 +137,4 @@
 
 	return fopencookie(cookiewrap, mode, funcswrap);
 }
-#else
-#error "Function funopen() needs to be ported."
 #endif
diff -Naur libbsd-0.8.3.orig/src/nlist.c libbsd-0.8.3/src/nlist.c
--- libbsd-0.8.3.orig/src/nlist.c	2016-08-10 13:54:09.063111767 -0700
+++ libbsd-0.8.3/src/nlist.c	2016-08-10 14:03:35.312927245 -0700
@@ -40,11 +40,12 @@
 
 #include <errno.h>
 #include <fcntl.h>
-#include <a.out.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 
+#include "nlist.h"
+
 #if !defined(__NO_A_OUT_SUPPORT)
 #define _NLIST_DO_AOUT
 #endif
