diff -Naur libbsd-0.8.2.orig/src/flopen.c libbsd-0.8.2/src/flopen.c
--- libbsd-0.8.2.orig/src/flopen.c	2011-07-05 09:56:44.000000000 -0700
+++ libbsd-0.8.2/src/flopen.c	2016-02-06 17:45:19.491982943 -0800
@@ -32,6 +32,7 @@
 #include <sys/stat.h>
 
 #include <errno.h>
+#include <fcntl.h>
 #include <stdarg.h>
 #include <unistd.h>
 #include <libutil.h>
diff -Naur libbsd-0.8.2.orig/src/funopen.c libbsd-0.8.2/src/funopen.c
--- libbsd-0.8.2.orig/src/funopen.c	2015-09-22 10:56:09.000000000 -0700
+++ libbsd-0.8.2/src/funopen.c	2016-02-06 17:45:38.954982595 -0800
@@ -137,6 +137,4 @@
 
 	return fopencookie(cookiewrap, mode, funcswrap);
 }
-#else
-#error "Function funopen() needs to be ported."
 #endif
diff -Naur libbsd-0.8.2.orig/src/nlist.c libbsd-0.8.2/src/nlist.c
--- libbsd-0.8.2.orig/src/nlist.c	2015-09-22 22:59:34.000000000 -0700
+++ libbsd-0.8.2/src/nlist.c	2016-02-06 17:46:01.065982199 -0800
@@ -39,11 +39,13 @@
 #include <arpa/inet.h>
 
 #include <errno.h>
-#include <a.out.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 
+#include "nlist.h"
+
 #if !defined(__NO_A_OUT_SUPPORT)
 #define _NLIST_DO_AOUT
 #endif
