diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/err.h libbsd-0.8.2/include/bsd/err.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/err.h	2015-12-13 23:58:51.000000000 +0000
+++ libbsd-0.8.2/include/bsd/err.h	2016-04-04 23:37:30.448000000 +0000
@@ -42,7 +42,10 @@
 
 #include <stdarg.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 void warnc(int code, const char *format, ...)
 	__printflike(2, 3);
 void vwarnc(int code, const char *format, va_list ap)
@@ -51,6 +54,9 @@
 	__printflike(3, 4);
 void verrc(int status, int code, const char *format, va_list ap)
 	__printflike(3, 0);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/libutil.h libbsd-0.8.2/include/bsd/libutil.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/libutil.h	2015-09-22 17:56:09.000000000 +0000
+++ libbsd-0.8.2/include/bsd/libutil.h	2016-04-04 23:37:42.336000000 +0000
@@ -53,7 +53,10 @@
 	ino_t	pf_ino;
 };
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 int humanize_number(char *buf, size_t len, int64_t bytes,
     const char *suffix, int scale, int flags);
 int expand_number(const char *_buf, uint64_t *_num);
@@ -66,7 +69,10 @@
 int pidfile_remove(struct pidfh *pfh);
 
 char   *fparseln(FILE *, size_t *, size_t *, const char[3], int);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 /* humanize_number(3) */
 #define HN_DECIMAL              0x01
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/md5.h libbsd-0.8.2/include/bsd/md5.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/md5.h	2015-12-13 23:58:59.000000000 +0000
+++ libbsd-0.8.2/include/bsd/md5.h	2016-04-04 23:37:50.496000000 +0000
@@ -30,7 +30,10 @@
 #include <sys/cdefs.h>
 #include <sys/types.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 void	 MD5Init(MD5_CTX *);
 void	 MD5Update(MD5_CTX *, const uint8_t *, size_t)
 		__attribute__((__bounded__(__string__,2,3)));
@@ -49,6 +52,9 @@
 char	*MD5Data(const uint8_t *, size_t, char *)
 		__attribute__((__bounded__(__string__,1,2)))
 		__attribute__((__bounded__(__minbytes__,3,MD5_DIGEST_STRING_LENGTH)));
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif /* _MD5_H_ */
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/nlist.h libbsd-0.8.2/include/bsd/nlist.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/nlist.h	2015-09-22 17:56:09.000000000 +0000
+++ libbsd-0.8.2/include/bsd/nlist.h	2016-04-04 22:47:44.656000000 +0000
@@ -30,8 +30,14 @@
 #include <sys/cdefs.h>
 #include <a.out.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern int nlist(const char *filename, struct nlist *list);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/readpassphrase.h libbsd-0.8.2/include/bsd/readpassphrase.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/readpassphrase.h	2011-07-05 18:58:44.000000000 +0000
+++ libbsd-0.8.2/include/bsd/readpassphrase.h	2016-04-04 23:38:19.276000000 +0000
@@ -34,8 +34,14 @@
 #include <sys/cdefs.h>
 #include <sys/types.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 char * readpassphrase(const char *, char *, size_t, int);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif /* !_READPASSPHRASE_H_ */
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/stdio.h libbsd-0.8.2/include/bsd/stdio.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/stdio.h	2015-09-23 05:59:34.000000000 +0000
+++ libbsd-0.8.2/include/bsd/stdio.h	2016-04-04 23:38:34.016000000 +0000
@@ -41,7 +41,10 @@
 #include <sys/cdefs.h>
 #include <sys/types.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 const char *fmtcheck(const char *, const char *);
 
 /* XXX: The function requires cooperation from the system libc to store the
@@ -69,7 +72,10 @@
 #define fwopen(cookie, fn) funopen(cookie, NULL, fn, NULL, NULL)
 
 int fpurge(FILE *fp);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/stdlib.h libbsd-0.8.2/include/bsd/stdlib.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/stdlib.h	2015-09-23 05:14:21.000000000 +0000
+++ libbsd-0.8.2/include/bsd/stdlib.h	2016-04-04 23:38:46.384000000 +0000
@@ -46,7 +46,10 @@
 #include <sys/stat.h>
 #include <stdint.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 uint32_t arc4random(void);
 void arc4random_stir(void);
 void arc4random_addrandom(u_char *dat, int datlen);
@@ -73,6 +76,9 @@
                    const char **errstr);
 
 char *getbsize(int *headerlenp, long *blocksizep);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/string.h libbsd-0.8.2/include/bsd/string.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/string.h	2015-09-24 03:30:23.000000000 +0000
+++ libbsd-0.8.2/include/bsd/string.h	2016-04-04 23:39:00.040000000 +0000
@@ -36,13 +36,19 @@
 #include <sys/cdefs.h>
 #include <sys/types.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 size_t strlcpy(char *dst, const char *src, size_t siz);
 size_t strlcat(char *dst, const char *src, size_t siz);
 char *strnstr(const char *str, const char *find, size_t str_len);
 void strmode(mode_t mode, char *str);
 
 void explicit_bzero(void *buf, size_t len);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/stringlist.h libbsd-0.8.2/include/bsd/stringlist.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/stringlist.h	2015-09-22 17:56:09.000000000 +0000
+++ libbsd-0.8.2/include/bsd/stringlist.h	2016-04-04 23:39:08.168000000 +0000
@@ -43,12 +43,18 @@
 	size_t	  sl_cur;
 } StringList;
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 StringList	*sl_init(void);
 int		 sl_add(StringList *, char *);
 void		 sl_free(StringList *, int);
 char		*sl_find(StringList *, const char *);
 int		 sl_delete(StringList *, const char *, int);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif /* _STRINGLIST_H */
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/unistd.h libbsd-0.8.2/include/bsd/unistd.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/unistd.h	2015-12-13 23:58:51.000000000 +0000
+++ libbsd-0.8.2/include/bsd/unistd.h	2016-04-04 23:39:19.816000000 +0000
@@ -45,7 +45,10 @@
 #define S_ISTXT S_ISVTX
 #endif
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern int optreset;
 
 #ifdef LIBBSD_OVERLAY
@@ -68,6 +71,9 @@
 	__printflike(1, 2);
 
 int getpeereid(int s, uid_t *euid, gid_t *egid);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/vis.h libbsd-0.8.2/include/bsd/vis.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/vis.h	2011-07-05 18:58:44.000000000 +0000
+++ libbsd-0.8.2/include/bsd/vis.h	2016-04-04 23:39:28.360000000 +0000
@@ -74,7 +74,10 @@
 
 #include <sys/cdefs.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 char	*vis(char *, int, int, int);
 int	strvis(char *, const char *, int);
 int	strvisx(char *, const char *, size_t, int);
@@ -83,6 +86,9 @@
 int	strunvisx(char *, const char *, int);
 ssize_t strnunvis(char *, const char *, size_t);
 int	unvis(char *, int, int *, int);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif /* !_VIS_H_ */
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/wchar.h libbsd-0.8.2/include/bsd/wchar.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/include/bsd/wchar.h	2015-09-22 17:56:09.000000000 +0000
+++ libbsd-0.8.2/include/bsd/wchar.h	2016-04-04 23:39:35.656000000 +0000
@@ -43,12 +43,18 @@
 #include <sys/cdefs.h>
 #include <sys/types.h>
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 wchar_t *fgetwln(FILE *stream, size_t *len);
 
 size_t wcslcat(wchar_t *dst, const wchar_t *src, size_t size);
 size_t wcslcpy(wchar_t *dst, const wchar_t *src, size_t size);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif
 #endif
diff -Naur /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/hash/sha512.h libbsd-0.8.2/src/hash/sha512.h
--- /var/tmp/portage/dev-libs/libbsd-0.8.2/work/libbsd-0.8.2/src/hash/sha512.h	2015-12-01 02:35:17.000000000 +0000
+++ libbsd-0.8.2/src/hash/sha512.h	2016-04-05 00:17:13.576000000 +0000
@@ -39,7 +39,10 @@
 	unsigned char buf[128];
 } SHA512_CTX;
 
-__BEGIN_DECLS
+/* __BEGIN_DECLS */
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 void	SHA512_Init(SHA512_CTX *);
 void	SHA512_Update(SHA512_CTX *, const void *, size_t);
@@ -48,6 +51,9 @@
 char   *SHA512_File(const char *, char *);
 char   *SHA512_FileChunk(const char *, char *, off_t, off_t);
 char   *SHA512_Data(const void *, unsigned int, char *);
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
+/* __END_DECLS */
 
 #endif /* !_SHA512_H_ */
