diff -Naur libbsd-0.8.7.orig/include/bsd/stdlib.h libbsd-0.8.7/include/bsd/stdlib.h
--- libbsd-0.8.7.orig/include/bsd/stdlib.h	2018-01-30 13:28:46.240353330 -0800
+++ libbsd-0.8.7/include/bsd/stdlib.h	2018-01-30 13:39:33.539341738 -0800
@@ -67,9 +67,7 @@
                const unsigned char *table, unsigned endbyte);
 
 void *reallocf(void *ptr, size_t size);
-#if defined(_GNU_SOURCE) && defined(__GLIBC__) && !__GLIBC_PREREQ(2, 26)
 void *reallocarray(void *ptr, size_t nmemb, size_t size);
-#endif
 
 long long strtonum(const char *nptr, long long minval, long long maxval,
                    const char **errstr);
diff -Naur libbsd-0.8.7.orig/include/bsd/string.h libbsd-0.8.7/include/bsd/string.h
--- libbsd-0.8.7.orig/include/bsd/string.h	2018-01-30 13:28:46.239353330 -0800
+++ libbsd-0.8.7/include/bsd/string.h	2018-01-30 13:39:27.108341853 -0800
@@ -42,9 +42,7 @@
 char *strnstr(const char *str, const char *find, size_t str_len);
 void strmode(mode_t mode, char *str);
 
-#if defined(_GNU_SOURCE) && defined(__GLIBC__) && !__GLIBC_PREREQ(2, 25)
 void explicit_bzero(void *buf, size_t len);
-#endif
 __END_DECLS
 
 #endif
