--- a/shash-0.2.6/src/errors.c
+++ b/shash-0.2.6/src/errors.c
@@ -1,4 +1,5 @@
 #include <defines.h>
+#include <stdio.h>
 
 extern int quiet;
 
--- a/shash-0.2.6/src/random.c
+++ b/shash-0.2.6/src/random.c
@@ -18,6 +18,8 @@
 #define DEFINES_H
 #include <defines.h>
 #endif
+#include <stdlib.h>
+#include <stdio.h>
 
 #ifdef HAVE_DEV_RANDOM
 
--- a/shash-0.2.6/src/extra.c
+++ b/shash-0.2.6/src/extra.c
@@ -1,4 +1,5 @@
 #include <defines.h>
+#include <strings.h>
 
 #ifdef HAVE_STAT
 
