diff -Naurw iputils-s20150815/rdisc.c iputils-s20150815.new/rdisc.c
--- iputils-s20150815/rdisc.c	2015-08-15 07:07:28.000000000 -0700
+++ iputils-s20150815.new/rdisc.c	2015-10-05 14:41:28.963384717 -0700
@@ -115,26 +115,7 @@
 
 #define MAXIFS 32
 
-#if defined(__GLIBC__) && __GLIBC__ < 2
-/* For router advertisement */
-struct icmp_ra
-{
-	unsigned char	icmp_type;		/* type of message, see below */
-	unsigned char	icmp_code;		/* type sub code */
-	unsigned short	icmp_cksum;		/* ones complement cksum of struct */
-	unsigned char	icmp_num_addrs;
-	unsigned char	icmp_wpa;		/* Words per address */
-	short 	icmp_lifetime;
-};
-
-struct icmp_ra_addr
-{
-	__u32	ira_addr;
-	__u32	ira_preference;
-};
-#else
 #define icmp_ra icmp
-#endif
 
 /* Router constants */
 #define	MAX_INITIAL_ADVERT_INTERVAL	16
