diff -Naur iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf.orig/ping.c iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf/ping.c
--- iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf.orig/ping.c	2017-10-16 01:44:15.000000000 -0700
+++ iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf/ping.c	2017-10-28 11:48:03.198321306 -0700
@@ -77,6 +77,10 @@
 #define	NROUTES		9		/* number of record route slots */
 #define TOS_MAX		255		/* 8-bit TOS field */
 
+#ifndef SCOPE_DELIMITER
+# define SCOPE_DELIMITER '%'
+#endif
+
 static int ts_type;
 static int nroute = 0;
 static __u32 route[10];
