diff -Naur iputils-s20121221.orig/ping6.c iputils-s20121221/ping6.c
--- iputils-s20121221.orig/ping6.c	2012-12-21 14:01:07.000000000 +0000
+++ iputils-s20121221/ping6.c	2015-07-19 00:50:45.254912002 +0000
@@ -80,6 +80,13 @@
 #include <stringprep.h>
 #endif
 
+#ifndef AI_IDN
+#define AI_IDN	0x0040
+#endif
+#ifndef NI_IDN
+#define NI_IDN	32
+#endif
+
 #include "ping6_niquery.h"
 #include "in6_flowlabel.h"
 
diff -Naur iputils-s20121221.orig/tracepath6.c iputils-s20121221/tracepath6.c
--- iputils-s20121221.orig/tracepath6.c	2012-12-21 14:01:07.000000000 +0000
+++ iputils-s20121221/tracepath6.c	2015-07-19 00:51:06.490911203 +0000
@@ -31,6 +31,13 @@
 #include <locale.h>
 #endif
 
+#ifndef AI_IDN
+#define AI_IDN  0x0040
+#endif
+#ifndef NI_IDN
+#define NI_IDN  32
+#endif
+
 #ifndef SOL_IPV6
 #define SOL_IPV6 IPPROTO_IPV6
 #endif
