From 9db06271ea75090b651e635004729d7b25b9a4f6 Mon Sep 17 00:00:00 2001
From: "Jory A. Pratt" <anarchy@gentoo.org>
Date: Mon, 4 Jan 2016 16:32:03 -0600
Subject: [PATCH] add missing limits.h include

Signed-off-by: Jory A. Pratt <anarchy@gentoo.org>
---
 reglib.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/reglib.c b/reglib.c
index 1fafd37..aa1b857 100644
--- a/reglib.c
+++ b/reglib.c
@@ -10,6 +10,7 @@
 #include <stdbool.h>
 #include <unistd.h>
 #include <string.h>
+#include <limits.h>
 
 #include <arpa/inet.h> /* ntohl */
 
-- 
2.6.4

