From a4f1f46b713ac468c65286d5d7d8cf5b55c05378 Mon Sep 17 00:00:00 2001
From: Jory Pratt <anarchy@gentoo.org>
Date: Tue, 10 Nov 2020 08:08:26 -0600
Subject: [PATCH 1/5] add missing stdint.h

---
 xbmc/filesystem/ZipManager.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/filesystem/ZipManager.h b/xbmc/filesystem/ZipManager.h
index 4575415..4da4b85 100644
--- a/xbmc/filesystem/ZipManager.h
+++ b/xbmc/filesystem/ZipManager.h
@@ -22,6 +22,7 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <stdint.h>
 
 class CURL;
 
-- 
2.26.2

