diff -Naur openmp-3.9.0.orig/runtime/src/kmp_str.c openmp-3.9.0/runtime/src/kmp_str.c
--- openmp-3.9.0.orig/runtime/src/kmp_str.c	2016-11-01 12:48:23.089878111 -0700
+++ openmp-3.9.0/runtime/src/kmp_str.c	2016-11-01 12:49:43.043419074 -0700
@@ -208,7 +208,7 @@
 
             #if ! KMP_OS_WINDOWS
                 va_list _args;
-                __va_copy( _args, args );  // Make copy of args.
+                va_copy( _args, args );  // Make copy of args.
                 #define args _args         // Substitute args with its copy, _args.
             #endif // KMP_OS_WINDOWS
             rc = KMP_VSNPRINTF( buffer->str + buffer->used, free, format, args );
