From cb8e8ab68bd2af477dc73c5ad5d0bc10bacab8f7 Mon Sep 17 00:00:00 2001
From: "Jory A. Pratt" <anarchy@gentoo.org>
Date: Sat, 9 Sep 2017 08:20:40 -0500
Subject: [PATCH] Use va_cpoy macro

Signed-off-by: Jory A. Pratt <anarchy@gentoo.org>
---
 runtime/src/kmp_str.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/src/kmp_str.cpp b/runtime/src/kmp_str.cpp
index 88967e1..75d33df 100644
--- a/runtime/src/kmp_str.cpp
+++ b/runtime/src/kmp_str.cpp
@@ -168,7 +168,7 @@ void __kmp_str_buf_vprint(kmp_str_buf_t *buffer, char const *format,
 
 #if !KMP_OS_WINDOWS
       va_list _args;
-      __va_copy(_args, args); // Make copy of args.
+      va_copy(_args, args); // Make copy of args.
 #define args _args // Substitute args with its copy, _args.
 #endif // KMP_OS_WINDOWS
       rc = KMP_VSNPRINTF(buffer->str + buffer->used, free, format, args);
-- 
2.14.1

