diff -Naur openmp-4.0.0.src.orig/runtime/src/kmp_str.cpp openmp-4.0.0.src/runtime/src/kmp_str.cpp
--- openmp-4.0.0.src.orig/runtime/src/kmp_str.cpp	2017-03-13 16:59:41.294879402 -0700
+++ openmp-4.0.0.src/runtime/src/kmp_str.cpp	2017-03-13 17:00:40.521420835 -0700
@@ -208,7 +208,7 @@
 
             #if ! KMP_OS_WINDOWS
                 va_list _args;
-                __va_copy( _args, args );  // Make copy of args.
+                va_copy( _args, args );  // Make copy of args.
                 #define args _args         // Substitute args with its copy, _args.
             #endif // KMP_OS_WINDOWS
             rc = KMP_VSNPRINTF( buffer->str + buffer->used, free, format, args );
