diff -Naur wayland-1.11.0.orig/src/scanner.c wayland-1.11.0/src/scanner.c
--- wayland-1.11.0.orig/src/scanner.c	2016-06-04 12:01:44.211512287 -0700
+++ wayland-1.11.0/src/scanner.c	2016-06-04 12:01:14.270164487 -0700
@@ -808,7 +808,7 @@
 	struct interface *i;
 	struct enumeration *e;
 	char *enum_name;
-	uint idx = 0, j;
+	uint32_t idx = 0, j;
 
 	for (j = 0; j + 1 < strlen(enum_attribute); j++) {
 		if (enum_attribute[j] == '.') {
