--- a/busybox-1.21.0/networking/libiproute/iplink.c
+++ b/busybox-1.21.0/networking/libiproute/iplink.c
@@ -5,8 +5,7 @@
  * Licensed under GPLv2 or later, see file LICENSE in this source tree.
  */
 #include <net/if.h>
-#include <net/if_packet.h>
-#include <netpacket/packet.h>
+#include <linux/if_packet.h>
 #include <netinet/if_ether.h>
 
 #include "ip_common.h"  /* #include "libbb.h" is inside */
