--- a/src/lisp.h	2015-02-03 20:42:39.233177373 -0100
+++ b/src/lisp.h	2015-02-03 20:44:01.780145394 -0100
@@ -270,6 +270,7 @@
    This can be used in #if, e.g., '#if VAL_MAX < UINTPTR_MAX' below.  */
 #define VAL_MAX (EMACS_INT_MAX >> (GCTYPEBITS - 1))
 
+#define USE_LSB_TAG 1
 /* Unless otherwise specified, use USE_LSB_TAG on systems where:  */
 #ifndef USE_LSB_TAG
 /* 1.  We know malloc returns a multiple of 8.  */
