diff -Naur chrony-3.2-pre1.orig/conf.c chrony-3.2-pre1/conf.c
--- chrony-3.2-pre1.orig/conf.c	2017-07-29 09:37:52.464925620 -0700
+++ chrony-3.2-pre1/conf.c	2017-07-29 09:39:11.161169295 -0700
@@ -42,6 +42,10 @@
 #include "cmdparse.h"
 #include "util.h"
 
+#if !defined(GLOB_NOMAGIC)
+#define GLOB_NOMAGIC 0
+#endif
+
 /* ================================================== */
 /* Forward prototypes */
 
