diff -Naur irrlicht-1.8.1.orig/source/Irrlicht/COSOperator.cpp irrlicht-1.8.1/source/Irrlicht/COSOperator.cpp
--- irrlicht-1.8.1.orig/source/Irrlicht/COSOperator.cpp	2016-08-14 18:44:42.231442788 -0700
+++ irrlicht-1.8.1/source/Irrlicht/COSOperator.cpp	2016-08-14 18:50:25.149101101 -0700
@@ -13,9 +13,11 @@
 #include <unistd.h>
 #ifndef _IRR_SOLARIS_PLATFORM_
 #include <sys/types.h>
+#ifndef __linux__
 #include <sys/sysctl.h>
 #endif
 #endif
+#endif
 
 #if defined(_IRR_COMPILE_WITH_X11_DEVICE_)
 #include "CIrrDeviceLinux.h"
