--- a/src/futils.cpp	2016-06-01 18:29:06.629998810 -0500
+++ b/src/futils.cpp	2016-06-01 18:29:22.209998806 -0500
@@ -332,8 +332,7 @@
         std::ostringstream os;
 #ifdef EXV_HAVE_STRERROR_R
         const size_t n = 1024;
-// _GNU_SOURCE: See Debian bug #485135
-# if defined EXV_STRERROR_R_CHAR_P && defined _GNU_SOURCE
+# if defined EXV_STRERROR_R_CHAR_P && defined __GLIBC__
         char *buf = 0;
         char buf2[n];
         std::memset(buf2, 0x0, n)
