diff -Naur netpbm-10.66.00.orig/converter/other/pstopnm.c netpbm-10.66.00/converter/other/pstopnm.c
--- netpbm-10.66.00.orig/converter/other/pstopnm.c	2014-01-01 15:59:14.000000000 -0500
+++ netpbm-10.66.00/converter/other/pstopnm.c	2015-07-18 17:52:11.501826392 -0400
@@ -79,7 +79,7 @@
     unsigned int ysize;  /* zero means unspecified */
     unsigned int dpi;    /* zero means unspecified */
     enum Orientation orientation;
-    unsigned int stdout;
+    unsigned int stdout1;
     unsigned int textalphabits;
 };
 
@@ -127,7 +127,7 @@
     OPTENT3(0, "dpi",        OPT_UINT,  &cmdlineP->dpi, &dpiSpec,        0);
     OPTENT3(0, "portrait",   OPT_FLAG,  NULL, &portraitOpt,             0);
     OPTENT3(0, "landscape",  OPT_FLAG,  NULL, &landscapeOpt,            0);
-    OPTENT3(0, "stdout",     OPT_FLAG,  NULL, &cmdlineP->stdout,         0);
+    OPTENT3(0, "stdout",     OPT_FLAG,  NULL, &cmdlineP->stdout1,         0);
     OPTENT3(0, "textalphabits", OPT_UINT,
             &cmdlineP->textalphabits,  &textalphabitsSpec, 0);
 
@@ -664,7 +664,7 @@
 -----------------------------------------------------------------------------*/
     const char * retval;  /* malloc'ed */
 
-    if (cmdline.stdout)
+    if (cmdline.stdout1)
         retval = strdup("-");
     else if (streq(cmdline.inputFileName, "-"))
         retval = strdup("-");
