--- a/base/logging.cpp
+++ b/base/logging.cpp
@@ -28,7 +28,7 @@
 // For getprogname(3) or program_invocation_short_name.
 #if defined(__ANDROID__) || defined(__APPLE__)
 #include <stdlib.h>
-#elif defined(__GLIBC__)
+#elif defined(__linux__)
 #include <errno.h>
 #endif
 
@@ -90,7 +90,7 @@
 }
 
 namespace {
-#if defined(__GLIBC__)
+#if defined(__linux__)
 const char* getprogname() {
   return program_invocation_short_name;
 }
--- a/adb/diagnose_usb.cpp	2018-01-29 12:05:10.466834056 -0800
+++ b/adb/diagnose_usb.cpp	2018-01-29 12:13:20.980825271 -0800
@@ -33,7 +33,7 @@
 // Returns a message describing any potential problems we find with udev, or an empty string if we
 // can't find plugdev information (i.e. udev is not installed).
 static std::string GetUdevProblem() {
-#if defined(__linux__)
+#if defined(__GLIBC__)
     errno = 0;
     group* plugdev_group = getgrnam("plugdev");
 
--- a/fastboot/socket.cpp	2018-01-29 12:05:10.493834055 -0800
+++ b/fastboot/socket.cpp	2018-01-29 12:17:23.666820925 -0800
@@ -28,6 +28,8 @@
 
 #include "socket.h"
 
+#include <sys/select.h>
+
 #include <android-base/errors.h>
 #include <android-base/stringprintf.h>
 
