diff -Naur android-tools.orig/core/include/utils/Compat.h android-tools/core/include/utils/Compat.h
--- android-tools.orig/core/include/utils/Compat.h	2017-04-24 11:27:39.697736014 -0700
+++ android-tools/core/include/utils/Compat.h	2017-04-24 11:45:24.930908371 -0700
@@ -41,6 +41,10 @@
 #define DEFFILEMODE 0666
 #endif /* _WIN32 */
 
+#if defined(__linux__) && !defined(__GLIBC__)
+#define DEFFILEMODE 0666
+#endif
+
 #if defined(_WIN32)
 #define ZD "%ld"
 #define ZD_TYPE long
