diff -Naur qtwebkit-opensource-src-5.5.1.orig/Source/JavaScriptCore/runtime/VMStackBounds.h qtwebkit-opensource-src-5.5.1/Source/JavaScriptCore/runtime/VMStackBounds.h
--- qtwebkit-opensource-src-5.5.1.orig/Source/JavaScriptCore/runtime/VMStackBounds.h	2016-02-27 04:22:08.000000000 -0800
+++ qtwebkit-opensource-src-5.5.1/Source/JavaScriptCore/runtime/VMStackBounds.h	2016-04-10 19:30:47.907626687 -0700
@@ -53,8 +53,8 @@
         //
         // These sizes were derived from the stack usage of a number of sites when
         // layout occurs when we've already consumed most of the C stack.
-        const size_t requiredStack = 128 * KB;
-        const size_t errorModeRequiredStack = 64 * KB;
+        const size_t requiredStack = 32 * KB;
+        const size_t errorModeRequiredStack = 16 * KB;
 
         size_t requiredCapacity = interpreter->isInErrorHandlingMode() ? errorModeRequiredStack : requiredStack;
         RELEASE_ASSERT(m_bounds.size() >= requiredCapacity);
