diff --git a/config/udev.c b/config/udev.c
index e198e8609..3a73189e2 100644
--- a/config/udev.c
+++ b/config/udev.c
@@ -135,7 +135,7 @@ device_added(struct udev_device *udev_device)
 #endif
 
     value = udev_device_get_property_value(udev_device, "ID_INPUT");
-    if (value && !strcmp(value, "0")) {
+    if (!value || !strcmp(value, "0")) {
         LogMessageVerb(X_INFO, 10,
                        "config/udev: ignoring device %s without "
                        "property ID_INPUT set\n", path);
