From 4d5c555b05cf09e186cab9411aea3155018dab59 Mon Sep 17 00:00:00 2001
From: 
Date: Fri, 30 Apr 2021 22:50:01 -0500
Subject: [PATCH 1/4] 

---
 hw/xfree86/loader/loadmod.c | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/hw/xfree86/loader/loadmod.c b/hw/xfree86/loader/loadmod.c
index a6356bd..a2130a0 100644
--- a/hw/xfree86/loader/loadmod.c
+++ b/hw/xfree86/loader/loadmod.c
@@ -892,19 +892,18 @@ RemoveChild(ModuleDescPtr child)
     parent = child->parent;
     if (parent->child == child) {
         parent->child = child->sib;
-        return;
     }
-
-    prevsib = parent->child;
-    mdp = prevsib->sib;
-    while (mdp && mdp != child) {
-        prevsib = mdp;
-        mdp = mdp->sib;
+    else {
+        prevsib = parent->child;
+        mdp = prevsib->sib;
+        while (mdp && mdp != child) {
+            prevsib = mdp;
+            mdp = mdp->sib;
+        }
+        if (mdp == child)
+            prevsib->sib = child->sib;
     }
-    if (mdp == child)
-        prevsib->sib = child->sib;
     child->sib = NULL;
-    return;
 }
 
 void
-- 
2.26.3

