From 995276d672ecebf01b7a36f5ea7532f02f7ec68e Mon Sep 17 00:00:00 2001
From: Jory Pratt <anarchy@gentoo.org>
Date: Wed, 24 Apr 2019 10:13:46 -0500
Subject: [PATCH 4/7] fix fortify sources

---
 xbmc/cores/DllLoader/exports/wrapper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/DllLoader/exports/wrapper.c b/xbmc/cores/DllLoader/exports/wrapper.c
index 05e6368..a2ce079 100644
--- a/xbmc/cores/DllLoader/exports/wrapper.c
+++ b/xbmc/cores/DllLoader/exports/wrapper.c
@@ -453,7 +453,7 @@ struct mntent *__wrap_getmntent(FILE *fp)
 // are actually #defines which are inlined when compiled with -O. Those defines
 // actually call __*chk (for example, __fread_chk). We need to bypass this whole
 // thing to actually call our wrapped functions.
-#if _FORTIFY_SOURCE > 1
+#if defined(__GLIBC__) && _FORTIFY_SOURCE > 1
 
 size_t __wrap___fread_chk(void * ptr, size_t ptrlen, size_t size, size_t n, FILE * stream)
 {
-- 
2.21.0

