Fixes segfault when launching autofs with debugging info

--- autofs-5.1.4/daemon/automount.c	2017-12-19 03:46:44.000000000 +0100
+++ autofs-5.1.4/daemon/automount.c	2018-10-28 12:45:28.388254819 +0100
@@ -2495,16 +2495,10 @@ int main(int argc, char *argv[])
 		macro_free_global_table();
 		exit(1);
 	}
-
-	info(logging, "Starting automounter version %s, master map %s",
-		version, master_list->name);
-	info(logging, "using kernel protocol version %d.%02d",
-		get_kver_major(), get_kver_minor());
-
-	status = pthread_key_create(&key_thread_stdenv_vars,
-				key_thread_stdenv_vars_destroy);
+	
+	status = pthread_key_create(&key_thread_attempt_id, free);
 	if (status) {
-		logerr("%s: failed to create thread data key for std env vars!",
+		logerr("%s: failed to create thread data key for attempt ID!",
 		       program);
 		master_kill(master_list);
 		res = write(start_pipefd[1], pst_stat, sizeof(*pst_stat));
@@ -2513,10 +2507,15 @@ int main(int argc, char *argv[])
 		macro_free_global_table();
 		exit(1);
 	}
+	info(logging, "Starting automounter version %s, master map %s",
+		version, master_list->name);
+	info(logging, "using kernel protocol version %d.%02d",
+		get_kver_major(), get_kver_minor());
 
-	status = pthread_key_create(&key_thread_attempt_id, free);
+	status = pthread_key_create(&key_thread_stdenv_vars,
+				key_thread_stdenv_vars_destroy);
 	if (status) {
-		logerr("%s: failed to create thread data key for attempt ID!",
+		logerr("%s: failed to create thread data key for std env vars!",
 		       program);
 		master_kill(master_list);
 		res = write(start_pipefd[1], pst_stat, sizeof(*pst_stat));
