diff -Naur INTEL-SA-00075-Linux-Detection-And-Mitigation-Tools-1.0.orig/INTEL-SA-00075.h INTEL-SA-00075-Linux-Detection-And-Mitigation-Tools-1.0/INTEL-SA-00075.h
--- INTEL-SA-00075-Linux-Detection-And-Mitigation-Tools-1.0.orig/INTEL-SA-00075.h	2017-06-20 12:45:46.000000000 -0700
+++ INTEL-SA-00075-Linux-Detection-And-Mitigation-Tools-1.0/INTEL-SA-00075.h	2017-06-26 17:27:03.403349522 -0700
@@ -72,7 +72,6 @@
 #include <errno.h>
 #include <stdint.h>
 #include <stdbool.h>
-#include <bits/wordsize.h>
 #include <linux/mei.h>
 
 #define DEFAULT_MEI_DEV_NODE "/dev/mei0"
