From 64716d37e2bed00e06fb3463214d979ca6b3e9c0 Mon Sep 17 00:00:00 2001
From: 
Date: Mon, 2 Jul 2018 18:22:29 -0500
Subject: [PATCH] canonicalize_file_name only avaliable on glibc systems

---
 lib/sysfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/sysfs.c b/lib/sysfs.c
index 1adb50f..c909d0c 100644
--- a/lib/sysfs.c
+++ b/lib/sysfs.c
@@ -133,7 +133,7 @@ sysfs_deref_link(struct pci_dev *d, char *link_name)
   sysfs_obj_name(d, "", path);
   strcat(path, rel_path);
 
-  return canonicalize_file_name(path);
+  return realpath(path, NULL);
 }
 
 static int
-- 
2.18.0

