diff -Naur fontforge-20150430.orig/configure.ac fontforge-20150430/configure.ac
--- fontforge-20150430.orig/configure.ac	2015-04-30 20:29:11.000000000 +0000
+++ fontforge-20150430/configure.ac	2015-08-06 07:36:50.535844085 +0000
@@ -377,6 +377,7 @@
 if test x"${i_do_want_iconv}" = xyes; then
    AC_CHECK_HEADERS([iconv.h])
 fi
+AC_CHECK_HEADERS([execinfo.h])
 gl_INIT
 
 #--------------------------------------------------------------------------
diff -Naur fontforge-20150430.orig/fontforge/cvundoes.c fontforge-20150430/fontforge/cvundoes.c
--- fontforge-20150430.orig/fontforge/cvundoes.c	2015-04-30 20:29:11.000000000 +0000
+++ fontforge-20150430/fontforge/cvundoes.c	2015-08-06 07:38:15.766082215 +0000
@@ -24,6 +24,7 @@
  * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+#include "config.h"
 #include "fontforgevw.h"
 #include "views.h"
 #include <math.h>
@@ -33,8 +34,8 @@
 #include "inc/gfile.h"
 #include "psfont.h"
 
-#if defined(__MINGW32__)||defined(__CYGWIN__)
-// no backtrace on windows yet
+#ifndef HAVE_EXECINFO_H
+// no backtrace available
 #else
     #include <execinfo.h>
 #endif

