diff -Naur ModemManager-1.4.8.orig/libqcdm/src/com.c ModemManager-1.4.8/libqcdm/src/com.c
--- ModemManager-1.4.8.orig/libqcdm/src/com.c	2015-01-20 21:54:08.000000000 +0000
+++ ModemManager-1.4.8/libqcdm/src/com.c	2015-07-18 15:13:40.044214647 +0000
@@ -23,6 +23,11 @@
 #include "com.h"
 #include "errors.h"
 
+/* Missing from musl */
+#ifndef XCASE
+#define XCASE	0000004
+#endif
+
 int
 qcdm_port_setup (int fd)
 {
diff -Naur ModemManager-1.4.8.orig/libwmc/src/com.c ModemManager-1.4.8/libwmc/src/com.c
--- ModemManager-1.4.8.orig/libwmc/src/com.c	2015-01-20 21:54:08.000000000 +0000
+++ ModemManager-1.4.8/libwmc/src/com.c	2015-07-18 15:13:00.584216132 +0000
@@ -23,6 +23,11 @@
 #include "com.h"
 #include "errors.h"
 
+/* Missing from musl */
+#ifndef XCASE
+#define XCASE	0000004
+#endif
+
 int
 wmc_port_setup (int fd)
 {
diff -Naur ModemManager-1.4.8.orig/src/mm-port-serial.c ModemManager-1.4.8/src/mm-port-serial.c
--- ModemManager-1.4.8.orig/src/mm-port-serial.c	2015-02-07 19:28:58.000000000 +0000
+++ ModemManager-1.4.8/src/mm-port-serial.c	2015-07-18 15:13:22.992215289 +0000
@@ -36,6 +36,11 @@
 #include "mm-port-serial.h"
 #include "mm-log.h"
 
+/* Missing from musl */
+#ifndef XCASE
+#define XCASE	0000004
+#endif
+
 static gboolean port_serial_queue_process          (gpointer data);
 static void     port_serial_schedule_queue_process (MMPortSerial *self,
                                                     guint timeout_ms);
