diff -Naur a/i18n/digitlst.cpp b/i18n/digitlst.cpp
--- a/i18n/digitlst.cpp	2016-12-20 14:01:00.067384046 -0800
+++ b/i18n/digitlst.cpp	2016-12-20 14:01:21.773589684 -0800
@@ -61,11 +61,7 @@
 #endif
 
 #if U_USE_STRTOD_L && !U_PLATFORM_USES_ONLY_WIN32_API
-# if U_PLATFORM == U_PF_CYGWIN
-#   include <locale.h>
-# else
-#   include <xlocale.h>
-# endif
+# include <locale.h>
 #endif
 
 // ***************************************************************************
