diff -Naur net-tools-1.60_p20161110235919.orig/iptunnel.c net-tools-1.60_p20161110235919/iptunnel.c
--- net-tools-1.60_p20161110235919.orig/iptunnel.c	2016-11-10 15:59:19.000000000 -0800
+++ net-tools-1.60_p20161110235919/iptunnel.c	2016-11-11 12:38:47.646508006 -0800
@@ -29,6 +29,7 @@
 #include <arpa/inet.h>
 #include <net/if.h>
 #include <net/if_arp.h>
+#include <netinet/ip.h>
 #include <linux/types.h>
 #include <linux/if_tunnel.h>
 
