diff -Naur alsa-lib-1.1.3.orig/src/ucm/parser.c alsa-lib-1.1.3/src/ucm/parser.c
--- alsa-lib-1.1.3.orig/src/ucm/parser.c	2016-12-22 15:21:20.756308973 -0800
+++ alsa-lib-1.1.3/src/ucm/parser.c	2016-12-22 15:21:51.996612346 -0800
@@ -32,6 +32,7 @@
 
 #include "ucm_local.h"
 #include <dirent.h>
+#include <limits.h>
 
 /** The name of the environment variable containing the UCM directory */
 #define ALSA_CONFIG_UCM_VAR "ALSA_CONFIG_UCM"
