diff -ur a/mach_dep.c b/gc-7.4.2/mach_dep.c
--- a/mach_dep.c	2014-06-03 08:08:02.000000000 -0200
+++ b/mach_dep.c	2015-04-10 10:54:48.500634001 -0200
@@ -218,7 +218,7 @@
 
 #   if defined(HAVE_PUSH_REGS)
       GC_push_regs();
-#   elif defined(UNIX_LIKE) && !defined(NO_GETCONTEXT)
+#   elif 0
       /* Older versions of Darwin seem to lack getcontext(). */
       /* ARM and MIPS Linux often doesn't support a real     */
       /* getcontext().                                       */
