diff -Naur qemu-2.8.0.orig/linux-user/syscall.c qemu-2.8.0/linux-user/syscall.c
--- qemu-2.8.0.orig/linux-user/syscall.c	2016-12-20 12:16:48.000000000 -0800
+++ qemu-2.8.0/linux-user/syscall.c	2017-01-03 14:11:45.195429181 -0800
@@ -117,6 +117,12 @@
 #ifndef CLONE_IO
 #define CLONE_IO                0x80000000      /* Clone io context */
 #endif
+#ifndef F_SHLCK
+#define F_SHLCK 8
+#endif
+#ifndef F_EXLCK
+#define F_EXLCK 4
+#endif
 
 /* We can't directly call the host clone syscall, because this will
  * badly confuse libc (breaking mutexes, for example). So we must
