diff -Naur samba-4.15.7.orig/source3/printing/samba-bgqd.c samba-4.15.7/source3/printing/samba-bgqd.c
--- samba-4.15.7.orig/source3/printing/samba-bgqd.c	2022-03-15 09:19:34.136201100 -0400
+++ samba-4.15.7/source3/printing/samba-bgqd.c	2022-07-19 13:27:56.069779404 -0400
@@ -15,6 +15,7 @@
  *  along with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <signal.h>
 #include "replace.h"
 #include "system/filesys.h"
 #include "lib/util/server_id.h"
