diff -Naur samba-4.3.9.orig/lib/replace/replace.c samba-4.3.9/lib/replace/replace.c
--- samba-4.3.9.orig/lib/replace/replace.c	2015-09-08 13:45:25.000000000 +0000
+++ samba-4.3.9/lib/replace/replace.c	2016-05-29 06:47:07.529796925 +0000
@@ -558,43 +558,6 @@
 #endif /* HAVE_STRTOLL */
 
 
-#ifndef HAVE_STRTOULL
-unsigned long long int rep_strtoull(const char *str, char **endptr, int base)
-{
-#ifdef HAVE_STRTOUQ
-	return strtouq(str, endptr, base);
-#elif defined(HAVE___STRTOULL) 
-	return __strtoull(str, endptr, base);
-#elif SIZEOF_LONG == SIZEOF_LONG_LONG
-	return (unsigned long long int) strtoul(str, endptr, base);
-#else
-# error "You need a strtoull function"
-#endif
-}
-#else
-#ifdef HAVE_BSD_STRTOLL
-#ifdef HAVE_STRTOUQ
-unsigned long long int rep_strtoull(const char *str, char **endptr, int base)
-{
-	unsigned long long int nb = strtouq(str, endptr, base);
-	/* In linux EINVAL is only returned if base is not ok */
-	if (errno == EINVAL) {
-		if (base == 0 || (base >1 && base <37)) {
-			/* Base was ok so it's because we were not
-			 * able to make the convertion.
-			 * Let's reset errno.
-			 */
-			errno = 0;
-		}
-	}
-	return nb;
-}
-#else
-#error "You need the strtouq function"
-#endif /* HAVE_STRTOUQ */
-#endif /* HAVE_BSD_STRTOLL */
-#endif /* HAVE_STRTOULL */
-
 #ifndef HAVE_SETENV
 int rep_setenv(const char *name, const char *value, int overwrite) 
 {
diff -Naur samba-4.3.9.orig/lib/replace/replace.h samba-4.3.9/lib/replace/replace.h
--- samba-4.3.9.orig/lib/replace/replace.h	2015-09-08 13:45:25.000000000 +0000
+++ samba-4.3.9/lib/replace/replace.h	2016-05-29 06:47:28.886795715 +0000
@@ -371,16 +371,6 @@
 #endif
 #endif
 
-#ifndef HAVE_STRTOULL
-#define strtoull rep_strtoull
-unsigned long long int rep_strtoull(const char *str, char **endptr, int base);
-#else
-#ifdef HAVE_BSD_STRTOLL /* yes, it's not HAVE_BSD_STRTOULL */
-#define strtoull rep_strtoull
-unsigned long long int rep_strtoull(const char *str, char **endptr, int base);
-#endif
-#endif
-
 #ifndef HAVE_FTRUNCATE
 #define ftruncate rep_ftruncate
 int rep_ftruncate(int,off_t);
