diff -Naur elogind-243.7.orig/src/basic/selinux-util.c elogind-243.7/src/basic/selinux-util.c
--- elogind-243.7.orig/src/basic/selinux-util.c	2020-03-12 20:33:40.000000000 +0100
+++ elogind-243.7/src/basic/selinux-util.c	2020-07-03 12:29:16.140181565 +0200
@@ -63,7 +63,6 @@
 
 #if HAVE_SELINUX
         usec_t before_timestamp, after_timestamp;
-        struct mallinfo before_mallinfo, after_mallinfo;
 
         if (label_hnd)
                 return 0;
@@ -71,7 +70,6 @@
         if (!mac_selinux_use())
                 return 0;
 
-        before_mallinfo = mallinfo();
         before_timestamp = now(CLOCK_MONOTONIC);
 
         label_hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
@@ -83,13 +81,9 @@
                 int l;
 
                 after_timestamp = now(CLOCK_MONOTONIC);
-                after_mallinfo = mallinfo();
 
-                l = after_mallinfo.uordblks > before_mallinfo.uordblks ? after_mallinfo.uordblks - before_mallinfo.uordblks : 0;
-
-                log_debug("Successfully loaded SELinux database in %s, size on heap is %iK.",
-                          format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0),
-                          (l+1023)/1024);
+                log_debug("Successfully loaded SELinux database in %s",
+                          format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0));
         }
 #endif
 
