diff -Naur elogind-246.9.2-r1/src/basic/selinux-util.c elogind-246.9.2/src/basic/selinux-util.c
--- a/src/basic/selinux-util.c	2021-03-13 17:08:40.496016258 -0500
+++ b/src/basic/selinux-util.c	2021-03-13 17:10:54.928016323 -0500
@@ -75,7 +75,6 @@
 int mac_selinux_init(void) {
 #if HAVE_SELINUX
         usec_t before_timestamp, after_timestamp;
-        struct mallinfo before_mallinfo, after_mallinfo;
         char timespan[FORMAT_TIMESPAN_MAX];
         int l;
 
@@ -87,7 +86,6 @@
         if (!mac_selinux_use())
                 return 0;
 
-        before_mallinfo = mallinfo();
         before_timestamp = now(CLOCK_MONOTONIC);
 
         label_hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
@@ -95,14 +93,9 @@
                 return log_enforcing_errno(errno, "Failed to initialize SELinux labeling handle: %m");
 
         after_timestamp = now(CLOCK_MONOTONIC);
-        after_mallinfo = mallinfo();
-
-        l = after_mallinfo.uordblks > before_mallinfo.uordblks ? after_mallinfo.uordblks - before_mallinfo.uordblks : 0;
-
-        log_debug("Successfully loaded SELinux database in %s, size on heap is %iK.",
-                  format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0),
-                  (l+1023)/1024);
 
+	log_debug("Successfully loaded SELinux database in %s",
+		format_timespan(timespan, sizeof(timespan), after_timestamp - before_timestamp, 0));
 #endif
         return 0;
 }
