diff -Naru qbittorrent-4.0.4.orig/src/app/main.cpp qbittorrent-4.0.4/src/app/main.cpp
--- qbittorrent-4.0.4.orig/src/app/main.cpp	2018-03-16 10:08:10.493478535 +0100
+++ qbittorrent-4.0.4/src/app/main.cpp	2018-03-16 10:12:21.919464628 +0100
@@ -57,8 +57,6 @@
 
 #ifdef Q_OS_UNIX
 #include <signal.h>
-#include <execinfo.h>
-#include "stacktrace.h"
 #endif // Q_OS_UNIX
 
 #ifdef STACKTRACE_WIN
@@ -308,7 +306,6 @@
     reportToUser(msg);
     reportToUser(sigName);
     reportToUser("\n");
-    print_stacktrace();  // unsafe
 #endif // !defined Q_OS_WIN && !defined Q_OS_HAIKU
 #ifdef STACKTRACE_WIN
     StraceDlg dlg;  // unsafe

