diff -Naur libopenraw-0.0.9.orig/lib/io/posix_io.c libopenraw-0.0.9/lib/io/posix_io.c
--- libopenraw-0.0.9.orig/lib/io/posix_io.c	2011-11-12 16:46:29.000000000 +0000
+++ libopenraw-0.0.9/lib/io/posix_io.c	2015-07-18 12:12:24.088623814 +0000
@@ -39,7 +39,7 @@
 };
 
 static IOFileRef posix_open(const char *path, int mode);
-static int posix_close(IOFileRef f);
+static int posix_close1(IOFileRef f);
 static int posix_seek(IOFileRef f, off_t offset, int whence);
 static int posix_read(IOFileRef f, void *buf, size_t count);
 static off_t posix_filesize(IOFileRef f);
@@ -49,7 +49,7 @@
 /** posix io methods instance. Constant. */
 struct io_methods posix_io_methods = {
 	&posix_open,
-	&posix_close,
+	&posix_close1,
 	&posix_seek,
 	&posix_read,
 	&posix_filesize,
@@ -85,7 +85,7 @@
 
 
 /** posix implementation for close() */
-static int posix_close(IOFileRef f)
+static int posix_close1(IOFileRef f)
 {
 	int retval = 0;
 	struct io_data_posix *data = (struct io_data_posix*)f->_private;
