--- a/src/gn/CMakeLists.txt
+++ b/src/gn/CMakeLists.txt
@@ -46,6 +46,7 @@ add_custom_command(
         --ld ${GN_LINKER}
         --ar ${CMAKE_AR}
         --qt-version "${QT_REPO_MODULE_VERSION}.qtwebengine.qt.io"
+	--no-static-libstdc++
         $<$<PLATFORM_ID:Darwin>:--isysroot>
         $<$<PLATFORM_ID:Darwin>:${CMAKE_OSX_SYSROOT}>
     COMMAND Ninja::ninja -C ${GN_BINARY_DIR}/$<CONFIG> ${GN_EXECUTABLE}

