diff -Naur qtwebengine-opensource-src-5.6.1.orig/src/3rdparty/chromium/third_party/ffmpeg/libavutil/autorename_libavutil_cpu.c qtwebengine-opensource-src-5.6.1/src/3rdparty/chromium/third_party/ffmpeg/libavutil/autorename_libavutil_cpu.c
--- qtwebengine-opensource-src-5.6.1.orig/src/3rdparty/chromium/third_party/ffmpeg/libavutil/autorename_libavutil_cpu.c	2016-07-28 18:44:47.582449126 -0700
+++ qtwebengine-opensource-src-5.6.1/src/3rdparty/chromium/third_party/ffmpeg/libavutil/autorename_libavutil_cpu.c	2016-07-29 12:05:34.463048187 -0700
@@ -38,7 +38,6 @@
 #include <sys/param.h>
 #endif
 #include <sys/types.h>
-#include <sys/sysctl.h>
 #endif
 #if HAVE_UNISTD_H
 #include <unistd.h>
