From ce3c2cee2d0da85dc1a1d9ebc8ebad8dde05c10a Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Wed, 20 Jul 2022 18:07:19 +0200
Subject: [PATCH 1/2] sandbox syscall x86_64

---
 .../sandbox/linux/system_headers/x86_64_linux_syscalls.h      | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/3rdparty/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h b/src/3rdparty/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h
index e618c6237..c1f6bccf6 100644
--- a/src/3rdparty/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h
+++ b/src/3rdparty/chromium/sandbox/linux/system_headers/x86_64_linux_syscalls.h
@@ -1350,6 +1350,10 @@
 #define __NR_rseq 334
 #endif
 
+#if !defined(__NR_membarrier)
+#define __NR_membarrier 324
+#endif
+
 #if !defined(__NR_pidfd_send_signal)
 #define __NR_pidfd_send_signal 424
 #endif
-- 
2.35.1

