--- a/src/3rdparty/gn/src/base/files/file.h
+++ b/src/3rdparty/gn/src/base/files/file.h
@@ -26,7 +26,7 @@ namespace base {
     defined(OS_ANDROID) && __ANDROID_API__ < 21
 typedef struct stat stat_wrapper_t;
 #elif defined(OS_POSIX) || defined(OS_FUCHSIA)
-typedef struct stat64 stat_wrapper_t;
+typedef struct stat stat_wrapper_t;
 #endif
 
 // Thin wrapper around an OS-level file.
--- a/src/3rdparty/gn/src/base/files/file_posix.cc
+++ b/src/3rdparty/gn/src/base/files/file_posix.cc
@@ -32,7 +32,7 @@ int CallFstat(int fd, stat_wrapper_t* sb
 }
 #else
 int CallFstat(int fd, stat_wrapper_t* sb) {
-  return fstat64(fd, sb);
+  return fstat(fd, sb);
 }
 #endif
 
--- a/src/3rdparty/gn/src/base/files/file_util_posix.cc
+++ b/src/3rdparty/gn/src/base/files/file_util_posix.cc
@@ -70,10 +70,10 @@ int CallLstat(const char* path, stat_wra
 }
 #else
 int CallStat(const char* path, stat_wrapper_t* sb) {
-  return stat64(path, sb);
+  return stat(path, sb);
 }
 int CallLstat(const char* path, stat_wrapper_t* sb) {
-  return lstat64(path, sb);
+  return lstat(path, sb);
 }
 #endif
 
