From 7e5e61297eb4bb6f3ff0d3e01a85beb09c0e8d52 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Thu, 23 Feb 2017 10:53:38 +0100
Subject: [PATCH 2/2] build: musl compatibility

---
 src/libqmi-glib/qmi-utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libqmi-glib/qmi-utils.c b/src/libqmi-glib/qmi-utils.c
index 29e5f22..810640c 100644
--- a/src/libqmi-glib/qmi-utils.c
+++ b/src/libqmi-glib/qmi-utils.c
@@ -740,7 +740,7 @@ __qmi_utils_get_driver (const gchar *cdc_wdm_path)
          *    /sys/bus/usb/drivers/qmi_wwan
          */
         tmp = g_strdup_printf ("/sys/class/%s/%s/device/driver", subsystems[i], device_basename);
-        path = canonicalize_file_name (tmp);
+        path = realpath (tmp, NULL);
         g_free (tmp);
 
         if (g_file_test (path, G_FILE_TEST_EXISTS))
-- 
2.11.1

