diff -Naurw valgrind-3.11.0.orig/configure.ac valgrind-3.11.0/configure.ac
--- valgrind-3.11.0.orig/configure.ac	2015-09-22 13:26:31.000000000 -0700
+++ valgrind-3.11.0/configure.ac	2015-09-26 11:54:13.570025359 -0700
@@ -1066,8 +1066,6 @@
 	;;
      2.0|2.1|*)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 or later,])
-	AC_MSG_ERROR([Darwin libc, Bionic libc or Solaris libc])
 	;;
 esac
 
diff -Naurw valgrind-3.11.0.orig/coregrind/vg_preloaded.c valgrind-3.11.0/coregrind/vg_preloaded.c
--- valgrind-3.11.0.orig/coregrind/vg_preloaded.c	2015-09-08 06:23:26.000000000 -0700
+++ valgrind-3.11.0/coregrind/vg_preloaded.c	2015-09-26 11:54:52.383359849 -0700
@@ -56,7 +56,7 @@
 void VG_NOTIFY_ON_LOAD(freeres)( void );
 void VG_NOTIFY_ON_LOAD(freeres)( void )
 {
-#  if !defined(__UCLIBC__) \
+#  if defined(__GLIBC__) \
       && !defined(VGPV_arm_linux_android) \
       && !defined(VGPV_x86_linux_android) \
       && !defined(VGPV_mips32_linux_android) \
diff -Naurw valgrind-3.11.0.orig/include/pub_tool_redir.h valgrind-3.11.0/include/pub_tool_redir.h
--- valgrind-3.11.0.orig/include/pub_tool_redir.h	2015-09-08 06:23:26.000000000 -0700
+++ valgrind-3.11.0/include/pub_tool_redir.h	2015-09-26 11:57:20.796697605 -0700
@@ -242,7 +242,7 @@
 /* --- Soname of the standard C library. --- */
 
 #if defined(VGO_linux) || defined(VGO_solaris)
-#  define  VG_Z_LIBC_SONAME  libcZdsoZa              // libc.so*
+#  define  VG_Z_LIBC_SONAME  libcZdZa              // libc.*
 
 #elif defined(VGO_darwin) && (DARWIN_VERS <= DARWIN_10_6)
 #  define  VG_Z_LIBC_SONAME  libSystemZdZaZddylib    // libSystem.*.dylib
@@ -274,7 +274,7 @@
 /* --- Soname of the pthreads library. --- */
 
 #if defined(VGO_linux)
-#  define  VG_Z_LIBPTHREAD_SONAME  libpthreadZdsoZd0     // libpthread.so.0
+#  define  VG_Z_LIBPTHREAD_SONAME  libcZdZa    // libc.*
 #elif defined(VGO_darwin)
 #  define  VG_Z_LIBPTHREAD_SONAME  libSystemZdZaZddylib  // libSystem.*.dylib
 #elif defined(VGO_solaris)
