diff -Naur attr-2.4.47.orig/include/attributes.h attr-2.4.47/include/attributes.h
--- attr-2.4.47.orig/include/attributes.h	2013-05-19 00:53:54.000000000 -0400
+++ attr-2.4.47/include/attributes.h	2016-05-28 09:34:04.743603033 -0400
@@ -22,6 +22,8 @@
 extern "C" {
 #endif
 
+#include <sys/types.h>
+
 /*
  *	An almost-IRIX-compatible extended attributes API
  *	(the IRIX attribute "list" operation is missing, added ATTR_SECURE).
