diff -Naurw attr-2.4.47.orig/include/xattr.h attr-2.4.47/include/xattr.h
--- attr-2.4.47.orig/include/xattr.h	2015-08-22 14:26:31.167584051 +0700
+++ attr-2.4.47/include/xattr.h	2015-08-22 14:32:25.714261283 +0700
@@ -31,7 +31,12 @@
 #define XATTR_REPLACE 0x2       /* set value, fail if attr does not exist */
 
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+#define __THROW throw()
+extern "C" {
+#else
+#define __THROW
+#endif
 
 extern int setxattr (const char *__path, const char *__name,
 		      const void *__value, size_t __size, int __flags) __THROW;
@@ -58,6 +63,8 @@
 extern int lremovexattr (const char *__path, const char *__name) __THROW;
 extern int fremovexattr (int __filedes,   const char *__name) __THROW;
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* __XATTR_H__ */
