diff -Naur libdrm-2.4.91.orig/tests/nouveau/threaded.c libdrm-2.4.91/tests/nouveau/threaded.c
--- libdrm-2.4.91.orig/tests/nouveau/threaded.c	2018-03-05 21:06:45.000000000 +0000
+++ libdrm-2.4.91/tests/nouveau/threaded.c	2018-05-07 19:25:46.371706548 +0000
@@ -24,7 +24,10 @@
 #  include "config.h"
 #endif
 
-#include <sys/ioctl.h>
+// Use glibc's function prototype for ioctl()
+#include <bits/ioctl.h>
+int ioctl (int, unsigned long, ...);
+
 #include <dlfcn.h>
 #include <fcntl.h>
 #include <stdio.h>
