From dc34852ecca82a19694178e8d346bf1f413a6e1e Mon Sep 17 00:00:00 2001
From: Jory Pratt <anarchy@gentoo.org>
Date: Sat, 1 Aug 2020 18:26:33 -0500
Subject: [PATCH] don't build noveau tests

---
 tests/meson.build | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/tests/meson.build b/tests/meson.build
index 196edbf..651826b 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -44,9 +44,6 @@ endif
 if with_etnaviv
   subdir('etnaviv')
 endif
-if with_nouveau
-  subdir('nouveau')
-endif
 
 drmsl = executable(
   'drmsl',
-- 
2.26.2

