From 04d5d990904df782a7a774827e0c860eac640272 Mon Sep 17 00:00:00 2001
From: Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
Date: Thu, 16 Apr 2015 14:19:57 +0000
Subject: [PATCH 2/2] Fix format string in check.c

---
 src/check.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/check.c b/src/check.c
index d8880a8..bb783e5 100644
--- a/src/check.c
+++ b/src/check.c
@@ -582,7 +582,7 @@ static int check_file(DOS_FS * fs, DOS_FILE * file)
 	if (!(file->dir_ent.attr & ATTR_DIR) && le32toh(file->dir_ent.size) <=
 	    (uint64_t)clusters * fs->cluster_size) {
 	    printf
-		("%s\n  File size is %u bytes, cluster chain length is > %lu "
+		("%s\n  File size is %u bytes, cluster chain length is > %llu "
 		 "bytes.\n  Truncating file to %u bytes.\n", path_name(file),
 		 le32toh(file->dir_ent.size),
 		 (uint64_t)clusters * fs->cluster_size,
-- 
2.3.5

