diff -Naur libraw1394-2.1.1.orig/src/fw.c libraw1394-2.1.1/src/fw.c
--- libraw1394-2.1.1.orig/src/fw.c	2015-04-28 16:01:21.000000000 -0400
+++ libraw1394-2.1.1/src/fw.c	2015-07-15 02:58:18.267818961 -0400
@@ -187,7 +187,7 @@
 
 static int
 handle_echo_pipe(raw1394handle_t handle,
-		 struct epoll_closure *ec, __uint32_t events)
+		 struct epoll_closure *ec, uint32_t events)
 {
 	quadlet_t value;
 
@@ -258,7 +258,7 @@
 
 static int
 handle_device_event(raw1394handle_t handle,
-		    struct epoll_closure *ec, __uint32_t events)
+		    struct epoll_closure *ec, uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	union fw_cdev_event *u;
@@ -421,7 +421,7 @@
 
 static int
 handle_inotify(raw1394handle_t handle, struct epoll_closure *ec,
-	       __uint32_t events)
+	       uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	struct inotify_event *event;
diff -Naur libraw1394-2.1.1.orig/src/fw.h libraw1394-2.1.1/src/fw.h
--- libraw1394-2.1.1.orig/src/fw.h	2015-04-28 16:39:11.000000000 -0400
+++ libraw1394-2.1.1/src/fw.h	2015-07-15 02:58:37.132818057 -0400
@@ -48,7 +48,7 @@
 
 struct epoll_closure {
 	int (*func)(raw1394handle_t handle,
-		    struct epoll_closure *closure, __uint32_t events);
+		    struct epoll_closure *closure, uint32_t events);
 };
 
 struct port {
diff -Naur libraw1394-2.1.1.orig/src/fw-iso.c libraw1394-2.1.1/src/fw-iso.c
--- libraw1394-2.1.1.orig/src/fw-iso.c	2015-04-28 16:16:16.000000000 -0400
+++ libraw1394-2.1.1/src/fw-iso.c	2015-07-15 02:58:30.395818380 -0400
@@ -303,7 +303,7 @@
 }
 
 static int handle_iso_event(raw1394handle_t handle,
-			    struct epoll_closure *closure, __uint32_t events)
+			    struct epoll_closure *closure, uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	struct fw_cdev_event_iso_interrupt *interrupt;
