diff -Naur ipsec-tools-0.8.2.orig/src/racoon/cftoken.c ipsec-tools-0.8.2/src/racoon/cftoken.c
--- ipsec-tools-0.8.2.orig/src/racoon/cftoken.c	2012-08-29 11:24:51.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/cftoken.c	2018-08-11 08:01:12.177136634 +0000
@@ -4753,7 +4753,7 @@
 		return -1;
 	}
 
-	if (glob(path, GLOB_TILDE, NULL, &incstack[incstackp].matches) != 0 ||
+	if (glob(path, 0, NULL, &incstack[incstackp].matches) != 0 ||
 	    incstack[incstackp].matches.gl_pathc == 0) {
 		plog(LLV_ERROR, LOCATION, NULL,
 			"glob found no matches for path \"%s\"\n", path);
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/cftoken.l ipsec-tools-0.8.2/src/racoon/cftoken.l
--- ipsec-tools-0.8.2.orig/src/racoon/cftoken.l	2012-08-29 08:42:24.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/cftoken.l	2018-08-11 08:01:21.512854300 +0000
@@ -725,7 +725,7 @@
 		return -1;
 	}
 
-	if (glob(path, GLOB_TILDE, NULL, &incstack[incstackp].matches) != 0 ||
+	if (glob(path, 0, NULL, &incstack[incstackp].matches) != 0 ||
 	    incstack[incstackp].matches.gl_pathc == 0) {
 		plog(LLV_ERROR, LOCATION, NULL,
 			"glob found no matches for path \"%s\"\n", path);
