diff -Naur ipsec-tools-0.8.2.orig/src/racoon/backupsa.c ipsec-tools-0.8.2/src/racoon/backupsa.c
--- ipsec-tools-0.8.2.orig/src/racoon/backupsa.c	2010-04-02 15:15:00.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/backupsa.c	2018-08-11 14:22:15.868343597 +0000
@@ -276,9 +276,9 @@
 		GETNEXTNUM(sa_args.a_keylen, strtoul);
 		GETNEXTNUM(sa_args.flags, strtoul);
 		GETNEXTNUM(sa_args.l_alloc, strtoul);
-		GETNEXTNUM(sa_args.l_bytes, strtouq);
-		GETNEXTNUM(sa_args.l_addtime, strtouq);
-		GETNEXTNUM(sa_args.l_usetime, strtouq);
+		GETNEXTNUM(sa_args.l_bytes, strtoull);
+		GETNEXTNUM(sa_args.l_addtime, strtoull);
+		GETNEXTNUM(sa_args.l_usetime, strtoull);
 		GETNEXTNUM(sa_args.seq, strtoul);
 
 #undef GETNEXTNUM
