diff -Naur ipsec-tools-0.8.2.orig/src/libipsec/ipsec_strerror.h ipsec-tools-0.8.2/src/libipsec/ipsec_strerror.h
--- ipsec-tools-0.8.2.orig/src/libipsec/ipsec_strerror.h	2006-09-09 16:22:09.000000000 +0000
+++ ipsec-tools-0.8.2/src/libipsec/ipsec_strerror.h	2018-08-11 06:56:43.378265279 +0000
@@ -34,6 +34,8 @@
 #ifndef _IPSEC_STRERROR_H
 #define _IPSEC_STRERROR_H
 
+#define __P(protos) protos
+
 extern int __ipsec_errcode;
 extern void __ipsec_set_strerror __P((const char *));
 
diff -Naur ipsec-tools-0.8.2.orig/src/libipsec/libpfkey.h ipsec-tools-0.8.2/src/libipsec/libpfkey.h
--- ipsec-tools-0.8.2.orig/src/libipsec/libpfkey.h	2012-08-23 11:10:45.000000000 +0000
+++ ipsec-tools-0.8.2/src/libipsec/libpfkey.h	2018-08-11 06:58:44.102604340 +0000
@@ -44,6 +44,8 @@
 #define PRIORITY_OFFSET_POSITIVE_MAX	0x3fffffff
 #define PRIORITY_OFFSET_NEGATIVE_MAX	0x40000000
 
+#define __P(protos) protos
+
 struct sadb_msg;
 extern void pfkey_sadump __P((struct sadb_msg *));
 extern void pfkey_sadump_withports __P((struct sadb_msg *));
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/gnuc.h ipsec-tools-0.8.2/src/racoon/gnuc.h
--- ipsec-tools-0.8.2.orig/src/racoon/gnuc.h	2006-09-09 16:22:09.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/gnuc.h	2018-08-11 06:54:00.543203861 +0000
@@ -3,14 +3,7 @@
 /* Id: gnuc.h,v 1.4 2004/11/18 15:14:44 ludvigm Exp */
 
 /* Define __P() macro, if necessary */
-#undef __P
-#ifndef __P
-#if __STDC__
 #define __P(protos) protos
-#else
-#define __P(protos) ()
-#endif
-#endif
 
 /* inline foo */
 #ifdef __GNUC__
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/logger.h ipsec-tools-0.8.2/src/racoon/logger.h
--- ipsec-tools-0.8.2.orig/src/racoon/logger.h	2006-09-09 16:22:09.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/logger.h	2018-08-11 07:04:12.320653159 +0000
@@ -42,6 +42,8 @@
 	char *fname;
 };
 
+#define __P(protos) protos
+
 extern struct log *log_open __P((size_t, char *));
 extern void log_add __P((struct log *, char *));
 extern int log_print __P((struct log *, char *));
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/misc.h ipsec-tools-0.8.2/src/racoon/misc.h
--- ipsec-tools-0.8.2.orig/src/racoon/misc.h	2008-07-15 00:47:09.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/misc.h	2018-08-11 07:01:29.029603601 +0000
@@ -42,6 +42,8 @@
 #define LOCATION        debug_location(__FILE__, __LINE__, NULL)
 #endif
 
+#define __P(protos) protos
+
 extern int racoon_hexdump __P((void *, size_t));
 extern char *bit2str __P((int, int));
 extern void *get_newbuf __P((void *, size_t));
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/missing/crypto/sha2/sha2.h ipsec-tools-0.8.2/src/racoon/missing/crypto/sha2/sha2.h
--- ipsec-tools-0.8.2.orig/src/racoon/missing/crypto/sha2/sha2.h	2006-09-09 16:22:36.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/missing/crypto/sha2/sha2.h	2018-08-11 07:04:57.967269426 +0000
@@ -119,6 +119,8 @@
 
 /*** SHA-256/384/512 Function Prototypes ******************************/
 
+#define __P(protos) protos
+
 #ifndef HAVE_SHA2_IN_SHA_H
 void SHA256_Init __P((SHA256_CTX *));
 void SHA256_Update __P((SHA256_CTX*, const u_int8_t*, size_t));
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/netdb_dnssec.h ipsec-tools-0.8.2/src/racoon/netdb_dnssec.h
--- ipsec-tools-0.8.2.orig/src/racoon/netdb_dnssec.h	2006-09-09 16:22:09.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/netdb_dnssec.h	2018-08-11 07:06:07.145172460 +0000
@@ -68,6 +68,8 @@
 	struct certinfo *ci_next;	/* next structure */
 };
 
+#define __P(protos) protos
+
 extern void freecertinfo __P((struct certinfo *));
 extern int getcertsbyname __P((char *, struct certinfo **));
 
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/plog.h ipsec-tools-0.8.2/src/racoon/plog.h
--- ipsec-tools-0.8.2.orig/src/racoon/plog.h	2007-10-02 09:47:40.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/plog.h	2018-08-11 07:06:55.163716961 +0000
@@ -63,6 +63,8 @@
 extern int f_foreground;
 extern int print_location;
 
+#define __P(protos) protos
+
 struct sockaddr;
 #define plog(pri, ...) \
 	do { \
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/str2val.h ipsec-tools-0.8.2/src/racoon/str2val.h
--- ipsec-tools-0.8.2.orig/src/racoon/str2val.h	2006-09-09 16:22:10.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/str2val.h	2018-08-11 07:05:31.530252028 +0000
@@ -34,6 +34,8 @@
 #ifndef _STR2VAL_H
 #define _STR2VAL_H
 
+#define __P(protos) protos
+
 extern caddr_t val2str __P((const char *, size_t));
 extern char *str2val __P((const char *, int, size_t *));
 
diff -Naur ipsec-tools-0.8.2.orig/src/racoon/vmbuf.h ipsec-tools-0.8.2/src/racoon/vmbuf.h
--- ipsec-tools-0.8.2.orig/src/racoon/vmbuf.h	2006-09-09 16:22:10.000000000 +0000
+++ ipsec-tools-0.8.2/src/racoon/vmbuf.h	2018-08-11 07:03:41.489587802 +0000
@@ -65,6 +65,8 @@
 #define vfree   vmbuf_free
 #endif
 
+#define __P(protos) protos
+
 extern vchar_t *vmalloc __P((size_t));
 extern vchar_t *vrealloc __P((vchar_t *, size_t));
 extern void vfree __P((vchar_t *));
diff -Naur ipsec-tools-0.8.2.orig/src/setkey/extern.h ipsec-tools-0.8.2/src/setkey/extern.h
--- ipsec-tools-0.8.2.orig/src/setkey/extern.h	2009-03-06 11:45:03.000000000 +0000
+++ ipsec-tools-0.8.2/src/setkey/extern.h	2018-08-11 06:59:26.933305604 +0000
@@ -1,6 +1,6 @@
 /*	$NetBSD: extern.h,v 1.5 2009/03/06 11:45:03 tteras Exp $	*/
 
-
+#define __P(protos) protos
 
 void parse_init __P((void));
 int parse __P((FILE **));
