--- src/libply/ply-command-parser.c.orig	2017-08-09 05:48:57.000000000 -0700
+++ src/libply/ply-command-parser.c	2018-03-29 00:05:04.820405271 -0700
@@ -731,14 +731,23 @@
                 /* next argument isn't ours, so treat it like an unqualified
                  * flag
                  */
-                if (rpmatch (argument) < 0) {
-                        option->result.as_boolean = true;
-                        return true;
-                }
-
-                option->result.as_boolean = (bool) rpmatch (argument);
-                ply_list_remove_node (arguments, node);
-                return true;
+		switch (argument[0]) {
+		case 'N':
+		case 'n':
+			option->result.as_boolean = (bool) 0;
+			ply_list_remove_node (arguments, node);
+			return true;
+			break;
+		case 'Y':
+		case 'y':
+			option->result.as_boolean = (bool) 1;
+			ply_list_remove_node (arguments, node);
+			return true;
+			break;
+		default:
+			option->result.as_boolean = true;
+			return true;
+		}

         case PLY_COMMAND_OPTION_TYPE_STRING:
                 option->result.as_string = strdup (argument);
