diff -Naur iproute2-4.10.0.orig/bridge/link.c iproute2-4.10.0/bridge/link.c
--- iproute2-4.10.0.orig/bridge/link.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/bridge/link.c	2017-02-21 12:17:02.219298749 -0800
@@ -3,10 +3,10 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <time.h>
+#include <net/if.h>
 #include <sys/socket.h>
 #include <sys/time.h>
 #include <netinet/in.h>
-#include <linux/if.h>
 #include <linux/if_bridge.h>
 #include <string.h>
 #include <stdbool.h>
diff -Naur iproute2-4.10.0.orig/include/linux/if_bridge.h iproute2-4.10.0/include/linux/if_bridge.h
--- iproute2-4.10.0.orig/include/linux/if_bridge.h	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/include/linux/if_bridge.h	2017-02-21 12:28:42.531792431 -0800
@@ -15,7 +15,6 @@
 
 #include <linux/types.h>
 #include <linux/if_ether.h>
-#include <linux/in6.h>
 
 #define SYSFS_BRIDGE_ATTR	"bridge"
 #define SYSFS_BRIDGE_FDB	"brforward"
diff -Naur iproute2-4.10.0.orig/include/linux/if_tunnel.h iproute2-4.10.0/include/linux/if_tunnel.h
--- iproute2-4.10.0.orig/include/linux/if_tunnel.h	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/include/linux/if_tunnel.h	2017-02-21 12:26:13.871203393 -0800
@@ -2,9 +2,7 @@
 #define _IF_TUNNEL_H_
 
 #include <linux/types.h>
-#include <linux/if.h>
 #include <linux/ip.h>
-#include <linux/in6.h>
 #include <asm/byteorder.h>
 
 
diff -Naur iproute2-4.10.0.orig/include/linux/l2tp.h iproute2-4.10.0/include/linux/l2tp.h
--- iproute2-4.10.0.orig/include/linux/l2tp.h	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/include/linux/l2tp.h	2017-02-21 12:17:02.220298759 -0800
@@ -9,8 +9,7 @@
 
 #include <linux/types.h>
 #include <linux/socket.h>
-#include <linux/in.h>
-#include <linux/in6.h>
+#include <netinet/in.h>
 
 #define IPPROTO_L2TP		115
 
diff -Naur iproute2-4.10.0.orig/include/linux/netfilter.h iproute2-4.10.0/include/linux/netfilter.h
--- iproute2-4.10.0.orig/include/linux/netfilter.h	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/include/linux/netfilter.h	2017-02-21 12:30:04.969673613 -0800
@@ -4,8 +4,6 @@
 #include <linux/types.h>
 
 #include <linux/sysctl.h>
-#include <linux/in.h>
-#include <linux/in6.h>
 
 /* Responses from hook functions. */
 #define NF_DROP 0
diff -Naur iproute2-4.10.0.orig/include/linux/xfrm.h iproute2-4.10.0/include/linux/xfrm.h
--- iproute2-4.10.0.orig/include/linux/xfrm.h	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/include/linux/xfrm.h	2017-02-21 12:27:48.169211348 -0800
@@ -1,7 +1,6 @@
 #ifndef _LINUX_XFRM_H
 #define _LINUX_XFRM_H
 
-#include <linux/in6.h>
 #include <linux/types.h>
 
 /* All of the structures in this file may not change size as they are
diff -Naur iproute2-4.10.0.orig/ip/ip6tunnel.c iproute2-4.10.0/ip/ip6tunnel.c
--- iproute2-4.10.0.orig/ip/ip6tunnel.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/ip6tunnel.c	2017-02-21 12:23:12.679266625 -0800
@@ -23,13 +23,13 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <net/if.h>
+#include <net/if_arp.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
 #include <linux/ip.h>
-#include <linux/if.h>
-#include <linux/if_arp.h>
 #include <linux/if_tunnel.h>
 #include <linux/ip6_tunnel.h>
 
diff -Naur iproute2-4.10.0.orig/ip/ipaddress.c iproute2-4.10.0/ip/ipaddress.c
--- iproute2-4.10.0.orig/ip/ipaddress.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/ipaddress.c	2017-02-21 12:17:02.221298770 -0800
@@ -16,18 +16,18 @@
 #include <syslog.h>
 #include <inttypes.h>
 #include <fcntl.h>
+#include <net/if.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <sys/param.h>
 #include <errno.h>
 #include <netinet/in.h>
+#include <netinet/if_ether.h>
 #include <arpa/inet.h>
 #include <string.h>
 #include <fnmatch.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
 #include <linux/sockios.h>
 #include <linux/net_namespace.h>
 
diff -Naur iproute2-4.10.0.orig/ip/ipl2tp.c iproute2-4.10.0/ip/ipl2tp.c
--- iproute2-4.10.0.orig/ip/ipl2tp.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/ipl2tp.c	2017-02-21 12:17:02.222298781 -0800
@@ -19,8 +19,6 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
-#include <linux/if_arp.h>
 #include <linux/ip.h>
 
 #include <linux/genetlink.h>
diff -Naur iproute2-4.10.0.orig/ip/ipmaddr.c iproute2-4.10.0/ip/ipmaddr.c
--- iproute2-4.10.0.orig/ip/ipmaddr.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/ipmaddr.c	2017-02-21 12:17:02.222298781 -0800
@@ -15,15 +15,13 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if.h>
-#include <linux/if_arp.h>
 #include <linux/sockios.h>
 
 #include "rt_names.h"
diff -Naur iproute2-4.10.0.orig/ip/ipmroute.c iproute2-4.10.0/ip/ipmroute.c
--- iproute2-4.10.0.orig/ip/ipmroute.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/ipmroute.c	2017-02-21 12:17:02.223298792 -0800
@@ -22,9 +22,6 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if.h>
-#include <linux/if_arp.h>
 #include <linux/sockios.h>
 
 #include <rt_names.h>
diff -Naur iproute2-4.10.0.orig/ip/iprule.c iproute2-4.10.0/ip/iprule.c
--- iproute2-4.10.0.orig/ip/iprule.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/iprule.c	2017-02-21 12:17:02.223298792 -0800
@@ -15,12 +15,12 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netinet/ip.h>
 #include <arpa/inet.h>
 #include <string.h>
-#include <linux/if.h>
 #include <linux/fib_rules.h>
 #include <errno.h>
 
diff -Naur iproute2-4.10.0.orig/ip/iptoken.c iproute2-4.10.0/ip/iptoken.c
--- iproute2-4.10.0.orig/ip/iptoken.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/iptoken.c	2017-02-21 12:17:02.224298802 -0800
@@ -16,12 +16,12 @@
 #include <syslog.h>
 #include <fcntl.h>
 #include <string.h>
+#include <net/if.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netinet/ip.h>
 #include <arpa/inet.h>
 #include <linux/types.h>
-#include <linux/if.h>
 
 #include "rt_names.h"
 #include "utils.h"
diff -Naur iproute2-4.10.0.orig/ip/iptuntap.c iproute2-4.10.0/ip/iptuntap.c
--- iproute2-4.10.0.orig/ip/iptuntap.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/iptuntap.c	2017-02-21 12:17:02.224298802 -0800
@@ -14,11 +14,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <net/if.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
 #include <linux/if_tun.h>
 #include <pwd.h>
 #include <grp.h>
diff -Naur iproute2-4.10.0.orig/ip/tcp_metrics.c iproute2-4.10.0/ip/tcp_metrics.c
--- iproute2-4.10.0.orig/ip/tcp_metrics.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/tcp_metrics.c	2017-02-21 12:17:02.224298802 -0800
@@ -17,7 +17,6 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
 
 #include <linux/genetlink.h>
 #include <linux/tcp_metrics.h>
diff -Naur iproute2-4.10.0.orig/ip/tunnel.c iproute2-4.10.0/ip/tunnel.c
--- iproute2-4.10.0.orig/ip/tunnel.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/ip/tunnel.c	2017-02-21 12:22:52.571051688 -0800
@@ -26,11 +26,11 @@
 #include <string.h>
 #include <unistd.h>
 #include <errno.h>
+#include <net/if.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <netinet/in.h>
-#include <linux/if.h>
 #include <linux/ip.h>
 #include <linux/if_tunnel.h>
 
diff -Naur iproute2-4.10.0.orig/lib/ll_addr.c iproute2-4.10.0/lib/ll_addr.c
--- iproute2-4.10.0.orig/lib/ll_addr.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/lib/ll_addr.c	2017-02-21 12:17:02.225298813 -0800
@@ -14,6 +14,7 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if_arp.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
@@ -21,8 +22,6 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
 #include <linux/sockios.h>
 
 #include "rt_names.h"
diff -Naur iproute2-4.10.0.orig/lib/ll_proto.c iproute2-4.10.0/lib/ll_proto.c
--- iproute2-4.10.0.orig/lib/ll_proto.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/lib/ll_proto.c	2017-02-21 12:17:02.225298813 -0800
@@ -14,15 +14,15 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if_arp.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <netinet/in.h>
+#include <netinet/if_ether.h>
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
 #include <linux/sockios.h>
 
 #include "utils.h"
diff -Naur iproute2-4.10.0.orig/lib/ll_types.c iproute2-4.10.0/lib/ll_types.c
--- iproute2-4.10.0.orig/lib/ll_types.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/lib/ll_types.c	2017-02-21 12:17:02.225298813 -0800
@@ -14,6 +14,7 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if_arp.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
@@ -21,8 +22,6 @@
 #include <arpa/inet.h>
 #include <string.h>
 
-#include <linux/netdevice.h>
-#include <linux/if_arp.h>
 #include <linux/sockios.h>
 
 #include "rt_names.h"
diff -Naur iproute2-4.10.0.orig/lib/utils.c iproute2-4.10.0/lib/utils.c
--- iproute2-4.10.0.orig/lib/utils.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/lib/utils.c	2017-02-21 12:17:02.225298813 -0800
@@ -17,6 +17,7 @@
 #include <syslog.h>
 #include <fcntl.h>
 #include <limits.h>
+#include <net/if_arp.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <string.h>
@@ -25,7 +26,6 @@
 #include <asm/types.h>
 #include <linux/pkt_sched.h>
 #include <linux/param.h>
-#include <linux/if_arp.h>
 #include <linux/mpls.h>
 #include <time.h>
 #include <sys/time.h>
diff -Naur iproute2-4.10.0.orig/misc/arpd.c iproute2-4.10.0/misc/arpd.c
--- iproute2-4.10.0.orig/misc/arpd.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/misc/arpd.c	2017-02-21 12:17:02.226298824 -0800
@@ -18,18 +18,18 @@
 #include <netdb.h>
 #include <db_185.h>
 #include <sys/ioctl.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <net/if.h>
+#include <net/if_arp.h>
 #include <sys/uio.h>
 #include <sys/socket.h>
 #include <sys/time.h>
 #include <time.h>
 #include <signal.h>
-#include <linux/if.h>
-#include <linux/if_ether.h>
-#include <linux/if_arp.h>
 #include <netinet/in.h>
+#include <netinet/if_ether.h>
 #include <arpa/inet.h>
 #include <linux/if_packet.h>
 #include <linux/filter.h>
diff -Naur iproute2-4.10.0.orig/misc/ifstat.c iproute2-4.10.0/misc/ifstat.c
--- iproute2-4.10.0.orig/misc/ifstat.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/misc/ifstat.c	2017-02-21 12:17:02.226298824 -0800
@@ -21,7 +21,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <signal.h>
diff -Naur iproute2-4.10.0.orig/misc/nstat.c iproute2-4.10.0/misc/nstat.c
--- iproute2-4.10.0.orig/misc/nstat.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/misc/nstat.c	2017-02-21 12:17:02.227298835 -0800
@@ -21,7 +21,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <signal.h>
diff -Naur iproute2-4.10.0.orig/misc/rtacct.c iproute2-4.10.0/misc/rtacct.c
--- iproute2-4.10.0.orig/misc/rtacct.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/misc/rtacct.c	2017-02-21 12:17:02.227298835 -0800
@@ -22,7 +22,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
diff -Naur iproute2-4.10.0.orig/misc/ss.c iproute2-4.10.0/misc/ss.c
--- iproute2-4.10.0.orig/misc/ss.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/misc/ss.c	2017-02-21 12:17:02.228298845 -0800
@@ -14,6 +14,7 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if_arp.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/uio.h>
@@ -39,7 +40,6 @@
 #include <linux/sock_diag.h>
 #include <linux/inet_diag.h>
 #include <linux/unix_diag.h>
-#include <linux/netdevice.h>	/* for MAX_ADDR_LEN */
 #include <linux/filter.h>
 #include <linux/packet_diag.h>
 #include <linux/netlink_diag.h>
diff -Naur iproute2-4.10.0.orig/tc/f_basic.c iproute2-4.10.0/tc/f_basic.c
--- iproute2-4.10.0.orig/tc/f_basic.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/tc/f_basic.c	2017-02-21 12:17:02.228298845 -0800
@@ -19,7 +19,6 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
-#include <linux/if.h>
 
 #include "utils.h"
 #include "tc_util.h"
diff -Naur iproute2-4.10.0.orig/tc/f_fw.c iproute2-4.10.0/tc/f_fw.c
--- iproute2-4.10.0.orig/tc/f_fw.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/tc/f_fw.c	2017-02-21 12:17:02.229298856 -0800
@@ -15,11 +15,11 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
-#include <linux/if.h> /* IFNAMSIZ */
 #include "utils.h"
 #include "tc_util.h"
 
diff -Naur iproute2-4.10.0.orig/tc/f_matchall.c iproute2-4.10.0/tc/f_matchall.c
--- iproute2-4.10.0.orig/tc/f_matchall.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/tc/f_matchall.c	2017-02-21 12:17:02.229298856 -0800
@@ -19,7 +19,6 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
-#include <linux/if.h>
 
 #include "utils.h"
 #include "tc_util.h"
diff -Naur iproute2-4.10.0.orig/tc/f_u32.c iproute2-4.10.0/tc/f_u32.c
--- iproute2-4.10.0.orig/tc/f_u32.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/tc/f_u32.c	2017-02-21 12:17:02.229298856 -0800
@@ -16,12 +16,12 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <fcntl.h>
+#include <net/if.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <netinet/if_ether.h>
 #include <arpa/inet.h>
 #include <string.h>
-#include <linux/if.h>
-#include <linux/if_ether.h>
 
 #include "utils.h"
 #include "tc_util.h"
diff -Naur iproute2-4.10.0.orig/tc/m_ife.c iproute2-4.10.0/tc/m_ife.c
--- iproute2-4.10.0.orig/tc/m_ife.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/tc/m_ife.c	2017-02-21 12:17:02.230298867 -0800
@@ -17,9 +17,9 @@
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <netinet/if_ether.h>
 #include <arpa/inet.h>
 #include <string.h>
-#include <linux/netdevice.h>
 
 #include "rt_names.h"
 #include "utils.h"
diff -Naur iproute2-4.10.0.orig/tc/m_xt.c iproute2-4.10.0/tc/m_xt.c
--- iproute2-4.10.0.orig/tc/m_xt.c	2017-02-20 08:47:52.000000000 -0800
+++ iproute2-4.10.0/tc/m_xt.c	2017-02-21 12:17:02.230298867 -0800
@@ -14,10 +14,9 @@
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-#include <net/if.h>
 #include <limits.h>
 #include <linux/netfilter.h>
-#include <linux/netfilter_ipv4/ip_tables.h>
+#include <linux/netfilter_ipv4.h>
 #include <xtables.h>
 #include "utils.h"
 #include "tc_util.h"
