From 7a72799513088762bef49b55438f3c42acc50ab2 Mon Sep 17 00:00:00 2001
From: 
Date: Fri, 23 Jun 2017 22:27:04 -0500
Subject: [PATCH] check-for-musl-libcs-max_align_t

---
 include/stddef.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/stddef.h b/include/stddef.h
index 8841bbe..faf8552 100644
--- a/include/stddef.h
+++ b/include/stddef.h
@@ -53,7 +53,8 @@ using std::nullptr_t;
 }
 
 // Re-use the compiler's <stddef.h> max_align_t where possible.
-#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T)
+#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T) && \
+    !defined(__DEFINED_max_align_t)
 typedef long double max_align_t;
 #endif
 
-- 
2.13.1

