diff -Naur v4l-utils-1.6.2.orig/lib/libv4lconvert/libv4lsyscall-priv.h v4l-utils-1.6.2/lib/libv4lconvert/libv4lsyscall-priv.h
--- v4l-utils-1.6.2.orig/lib/libv4lconvert/libv4lsyscall-priv.h	2013-06-28 20:10:56.000000000 +0000
+++ v4l-utils-1.6.2/lib/libv4lconvert/libv4lsyscall-priv.h	2015-07-18 13:41:01.904423751 +0000
@@ -59,7 +59,12 @@
 #define	_IOC_SIZE(cmd) IOCPARM_LEN(cmd)
 #define	MAP_ANONYMOUS MAP_ANON
 #define	MMAP2_PAGE_SHIFT 0
+#endif
+
+#if !defined(__GLIBC__) && !defined(__UCLIBC__)
+#include <sys/types.h>
 typedef off_t __off_t;
+typedef off64_t __off64_t;
 #endif
 
 #undef SYS_OPEN
