--- a/v4l-utils-1.0.0/lib/libv4lconvert/libv4lsyscall-priv.h
+++ b/v4l-utils-1.0.0/lib/libv4lconvert/libv4lsyscall-priv.h
@@ -99,7 +99,7 @@
 #else
 #define SYS_MMAP(addr, len, prot, flags, fd, off) \
 	syscall(SYS_mmap2, (void *)(addr), (size_t)(len), \
-			(int)(prot), (int)(flags), (int)(fd), (__off_t)((off) >> MMAP2_PAGE_SHIFT))
+			(int)(prot), (int)(flags), (int)(fd), (off_t)((off) >> MMAP2_PAGE_SHIFT))
 #endif
 
 #define SYS_MUNMAP(addr, len) \
