From 411f7a1d30f03cb64e4716977e3504e25af8ce90 Mon Sep 17 00:00:00 2001
From: 
Date: Thu, 12 Dec 2019 21:58:31 -0600
Subject: [PATCH 8/9] musl strndupa

---
 src/unstrip.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/unstrip.c b/src/unstrip.c
index 9b8c09a..afce497 100644
--- a/src/unstrip.c
+++ b/src/unstrip.c
@@ -56,6 +56,15 @@
 # define _(str) gettext (str)
 #endif
 
+#ifndef strndupa
+#define strndupa(s, n) \
+       (__extension__ ({const char *__in = (s);                    \
+                        size_t __len = strnlen (__in, (n)) + 1;    \
+                        char *__out = (char *) alloca (__len);     \
+                        __out[__len-1] = '\0';                     \
+                        (char *) memcpy (__out, __in, __len-1);}))
+#endif
+
 /* Name and version of program.  */
 ARGP_PROGRAM_VERSION_HOOK_DEF = print_version;
 
-- 
2.24.1

