From fa623f8330c8e7264eba98d0e8199ef4aa219d2a Mon Sep 17 00:00:00 2001
From: 
Date: Thu, 12 Dec 2019 21:34:57 -0600
Subject: [PATCH 2/9] include asm.h for non glibc systems on ppc

---
 backends/ppc_initreg.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/backends/ppc_initreg.c b/backends/ppc_initreg.c
index 0e0d359..3dbcc2d 100644
--- a/backends/ppc_initreg.c
+++ b/backends/ppc_initreg.c
@@ -32,6 +32,9 @@
 
 #include <stdlib.h>
 #if defined(__powerpc__) && defined(__linux__)
+#if !defined(__GLIBC__)
+# include <asm/ptrace.h>
+#endif
 # include <sys/ptrace.h>
 # include <sys/user.h>
 #endif
-- 
2.24.1

