From 00bd0b5e75bbed231aa743502df96e0641da86fd Mon Sep 17 00:00:00 2001
From: 
Date: Thu, 12 Dec 2019 21:58:10 -0600
Subject: [PATCH 7/9] musl strerror_r

---
 libdwfl/dwfl_error.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libdwfl/dwfl_error.c b/libdwfl/dwfl_error.c
index 7bcf61c..ff25ebb 100644
--- a/libdwfl/dwfl_error.c
+++ b/libdwfl/dwfl_error.c
@@ -154,7 +154,16 @@ dwfl_errmsg (int error)
   switch (error &~ 0xffff)
     {
     case OTHER_ERROR (ERRNO):
+#if defined(__GLIBC__)
       return strerror_r (error & 0xffff, "bad", 0);
+#else
+      {
+        static __thread char buf[128] = "";
+        if (0 == strerror_r(error & 0xffff, buf, sizeof(buf)))
+          return buf;
+      }
+      return "strerror_r() failed";
+#endif
     case OTHER_ERROR (LIBELF):
       return elf_errmsg (error & 0xffff);
     case OTHER_ERROR (LIBDW):
-- 
2.24.1

