diff -Naur sandbox-2.10.orig/scripts/gen_symbol_version_map.awk sandbox-2.10/scripts/gen_symbol_version_map.awk
--- sandbox-2.10.orig/scripts/gen_symbol_version_map.awk	2009-08-26 01:59:48.000000000 -0400
+++ sandbox-2.10/scripts/gen_symbol_version_map.awk	2016-01-22 23:32:41.602858004 -0500
@@ -14,7 +14,7 @@
 
 	# Only check FUNCtion symbols which are not LOCAL, or
 	# do not have DEFAULT visibility
-	if ($4 != "FUNC" || $5 == "LOCAL" || $6 != "DEFAULT")
+	if ($4 != "FUNC" || $5 == "LOCAL" || ($6 != "DEFAULT" && $6 != "PROTECTED"))
 		next;
 
 	for (x in SYMBOLS) {
