diff -Naur zvbi-0.2.35.orig/src/misc.h zvbi-0.2.35/src/misc.h
--- zvbi-0.2.35.orig/src/misc.h	2013-07-01 19:32:31.000000000 -0700
+++ zvbi-0.2.35/src/misc.h	2016-05-02 19:15:03.874251498 -0700
@@ -425,7 +425,7 @@
 
 /* __va_copy is a GNU extension. */
 #ifndef __va_copy
-#  define __va_copy(ap1, ap2) do { ap1 = ap2; } while (0)
+#  define __va_copy(ap1, ap2) va_copy(ap1, ap2)
 #endif
 
 /* Use this instead of strncpy(). strlcpy() is a BSD extension. */
