From 523562d2c1a485fc60fe4cb5c2c02c5654c47097 Mon Sep 17 00:00:00 2001
From: 
Date: Wed, 10 Apr 2019 00:45:56 -0500
Subject: [PATCH 2/3] include sys/resource.h for RLIMIT_NOFILE

---
 modules/pam_unix/pam_unix_acct.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/pam_unix/pam_unix_acct.c b/modules/pam_unix/pam_unix_acct.c
index 8833114..eeab34e 100644
--- a/modules/pam_unix/pam_unix_acct.c
+++ b/modules/pam_unix/pam_unix_acct.c
@@ -48,6 +48,7 @@
 #include <time.h>		/* for time() */
 #include <errno.h>
 #include <sys/wait.h>
+#include <sys/resource.h>	/* for RLIMIT_NOFILE */
 
 #include <security/_pam_macros.h>
 
-- 
2.21.0

