diff --git a/build/moz.configure/rust.configure b/build/moz.configure/rust.configure
index b9d24654f8..ca7bf4baeb 100644
--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -262,6 +262,7 @@ def rust_supported_targets(rustc):
     return data
 
 
+@imports('os')
 def detect_rustc_target(
     host_or_target, compiler_info, arm_target, rust_supported_targets
 ):
@@ -274,9 +275,15 @@ def detect_rustc_target(
     # munging to get the correct option to rustc.
     # We correlate the autoconf-derived targets with the list of targets
     # rustc gives us with --print target-list.
-    candidates = rust_supported_targets.get(
-        (host_or_target.cpu, host_or_target.endianness, host_or_target.os), []
-    )
+    target_from_env = os.environ.get("RUST_TARGET")
+    if target_from_env is not None:
+        candidates = []
+        info = split_triplet(target_from_env, allow_wasi=True)
+        candidates.append(namespace(rust_target=target_from_env, target=info))
+    else:
+        candidates = rust_supported_targets.get(
+            (host_or_target.cpu, host_or_target.endianness, host_or_target.os), []
+        )
 
     def find_candidate(candidates):
         if len(candidates) == 1: