diff -Naur webkitgtk-2.4.1.orig/Source/WTF/wtf/Assertions.cpp webkitgtk-2.4.1/Source/WTF/wtf/Assertions.cpp
--- webkitgtk-2.4.1.orig/Source/WTF/wtf/Assertions.cpp	2014-04-14 02:40:45.000000000 -0400
+++ webkitgtk-2.4.1/Source/WTF/wtf/Assertions.cpp	2014-06-18 09:16:35.617334849 -0400
@@ -63,12 +63,6 @@
 #include <windows.h>
 #endif
 
-#if OS(DARWIN) || (OS(LINUX) && !defined(__UCLIBC__))
-#include <cxxabi.h>
-#include <dlfcn.h>
-#include <execinfo.h>
-#endif
-
 extern "C" {
 
 WTF_ATTRIBUTE_PRINTF(1, 0)
@@ -233,9 +227,7 @@
 
 void WTFGetBacktrace(void** stack, int* size)
 {
-#if OS(DARWIN) || (OS(LINUX) && !defined(__UCLIBC__))
-    *size = backtrace(stack, *size);
-#elif OS(WINDOWS) && !OS(WINCE)
+#if OS(WINDOWS) && !OS(WINCE)
     // The CaptureStackBackTrace function is available in XP, but it is not defined
     // in the Windows Server 2003 R2 Platform SDK. So, we'll grab the function
     // through GetProcAddress.
@@ -269,9 +261,6 @@
 
 #if OS(DARWIN) || OS(LINUX)
 #  if PLATFORM(GTK)
-#    if defined(__GLIBC__) && !defined(__UCLIBC__)
-#      define WTF_USE_BACKTRACE_SYMBOLS 1
-#    endif
 #  else
 #    define WTF_USE_DLADDR 1
 #  endif
