diff -Naur webkitgtk-2.4.9.orig/Source/WTF/wtf/DisallowCType.h webkitgtk-2.4.9/Source/WTF/wtf/DisallowCType.h
--- webkitgtk-2.4.9.orig/Source/WTF/wtf/DisallowCType.h	2015-05-20 05:03:24.000000000 -0400
+++ webkitgtk-2.4.9/Source/WTF/wtf/DisallowCType.h	2015-07-19 14:44:26.855225521 -0400
@@ -29,6 +29,8 @@
 #ifndef WTF_DisallowCType_h
 #define WTF_DisallowCType_h
 
+#if 0 // BRAIN DEAD - The detection here makes some bad assumptions.
+
 // The behavior of many of the functions in the <ctype.h> header is dependent
 // on the current locale. But almost all uses of these functions are for
 // locale-independent, ASCII-specific purposes. In WebKit code we use our own
@@ -81,4 +83,6 @@
 
 #endif
 
+#endif // BRAIN DEAD
+
 #endif
