diff -Naur webkitgtk-2.10.7.orig/Source/WTF/wtf/Assertions.cpp webkitgtk-2.10.7/Source/WTF/wtf/Assertions.cpp
--- webkitgtk-2.10.7.orig/Source/WTF/wtf/Assertions.cpp	2016-01-29 16:29:32.000000000 +0000
+++ webkitgtk-2.10.7/Source/WTF/wtf/Assertions.cpp	2016-03-15 00:12:11.432000000 +0000
@@ -71,7 +71,8 @@
 #if OS(DARWIN) || (OS(LINUX) && !defined(__UCLIBC__))
 #include <cxxabi.h>
 #include <dlfcn.h>
-#include <execinfo.h>
+// removed for musl - we need better intelligence here
+//#include <execinfo.h>
 #endif
 
 extern "C" {
@@ -226,7 +227,8 @@
 void WTFGetBacktrace(void** stack, int* size)
 {
 #if OS(DARWIN) || (OS(LINUX) && !defined(__UCLIBC__))
-    *size = backtrace(stack, *size);
+// removed for musl - we need better intelligence here
+//    *size = backtrace(stack, *size);
 #elif OS(WINDOWS)
     // The CaptureStackBackTrace function is available in XP, but it is not defined
     // in the Windows Server 2003 R2 Platform SDK. So, we'll grab the function
@@ -272,9 +274,10 @@
 void WTFPrintBacktrace(void** stack, int size)
 {
 #if USE(BACKTRACE_SYMBOLS)
-    char** symbols = backtrace_symbols(stack, size);
-    if (!symbols)
-        return;
+// removed for musl - we need better intelligence here
+//    char** symbols = backtrace_symbols(stack, size);
+//    if (!symbols)
+//        return;
 #endif
 
     for (int i = 0; i < size; ++i) {
