diff -Naur webkitgtk-2.10.9.orig/Source/WTF/wtf/Assertions.cpp webkitgtk-2.10.9/Source/WTF/wtf/Assertions.cpp
--- webkitgtk-2.10.9.orig/Source/WTF/wtf/Assertions.cpp	2016-03-17 02:48:18.000000000 -0400
+++ webkitgtk-2.10.9/Source/WTF/wtf/Assertions.cpp	2016-06-01 09:31:42.345411637 -0400
@@ -71,7 +71,7 @@
 #if OS(DARWIN) || (OS(LINUX) && defined(__GLIBC__) && !defined(__UCLIBC__))
 #include <cxxabi.h>
 #include <dlfcn.h>
-#include <execinfo.h>
+//#include <execinfo.h>
 #endif
 
 extern "C" {
@@ -226,7 +226,7 @@
 void WTFGetBacktrace(void** stack, int* size)
 {
 #if OS(DARWIN) || (OS(LINUX) && defined(__GLIBC__) && !defined(__UCLIBC__))
-    *size = backtrace(stack, *size);
+//    *size = backtrace(stack, *size);
 #elif OS(WINDOWS)
     // The CaptureStackBackTrace function is available in XP, but it is not defined
     // in the Windows Server 2003 R2 Platform SDK. So, we'll grab the function
@@ -272,9 +272,9 @@
 void WTFPrintBacktrace(void** stack, int size)
 {
 #if USE(BACKTRACE_SYMBOLS)
-    char** symbols = backtrace_symbols(stack, size);
-    if (!symbols)
-        return;
+//    char** symbols = backtrace_symbols(stack, size);
+//    if (!symbols)
+//        return;
 #endif
 
     for (int i = 0; i < size; ++i) {
