--- openjdk/jdk/src/solaris/native/sun/awt/awt_InputMethod.c
+++ openjdk/jdk/src/solaris/native/sun/awt/awt_InputMethod.c
@@ -1239,7 +1239,11 @@ StatusDrawCallback(XIC ic, XPointer client_data,
           }
           else{
               char *mbstr = wcstombsdmp(text->string.wide_char, text->length);
+              if (mbstr == NULL) {
+                  goto finally;
+              }
               strcpy(statusWindow->status, mbstr);
+              free(mbstr);
           }
           statusWindow->on = True;
           onoffStatusWindow(pX11IMData, statusWindow->parent, True);
