diff -Naur openjdk.orig/jdk/src/aix/native/java/net/aix_close.c openjdk/jdk/src/aix/native/java/net/aix_close.c
--- openjdk.orig/jdk/src/aix/native/java/net/aix_close.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/aix/native/java/net/aix_close.c	2020-07-11 11:27:09.067892057 +0200
@@ -65,7 +65,7 @@
 #include <unistd.h>
 #include <errno.h>
 
-#include <sys/poll.h>
+#include <poll.h>
 
 /*
  * Stack allocated by thread when doing blocking operation
diff -Naur openjdk.orig/jdk/src/aix/native/sun/nio/ch/AixNativeThread.c openjdk/jdk/src/aix/native/sun/nio/ch/AixNativeThread.c
--- openjdk.orig/jdk/src/aix/native/sun/nio/ch/AixNativeThread.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/aix/native/sun/nio/ch/AixNativeThread.c	2020-07-11 11:27:26.311893251 +0200
@@ -32,7 +32,7 @@
 #include "sun_nio_ch_NativeThread.h"
 
 #include <pthread.h>
-#include <sys/signal.h>
+#include <signal.h>
 
 /* Also defined in src/aix/native/java/net/aix_close.c */
 #define INTERRUPT_SIGNAL (SIGRTMAX - 1)
diff -Naur openjdk.orig/jdk/src/aix/native/sun/nio/ch/AixPollPort.c openjdk/jdk/src/aix/native/sun/nio/ch/AixPollPort.c
--- openjdk.orig/jdk/src/aix/native/sun/nio/ch/AixPollPort.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/aix/native/sun/nio/ch/AixPollPort.c	2020-07-11 11:27:44.675894522 +0200
@@ -34,7 +34,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/pollset.h>
 #include <fcntl.h>
 #include <stddef.h>
diff -Naur openjdk.orig/jdk/src/macosx/javavm/export/jvm_md.h openjdk/jdk/src/macosx/javavm/export/jvm_md.h
--- openjdk.orig/jdk/src/macosx/javavm/export/jvm_md.h	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/macosx/javavm/export/jvm_md.h	2020-07-11 11:28:01.659895698 +0200
@@ -60,7 +60,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <sys/signal.h>
+#include <signal.h>
 
 /* O Flags */
 
diff -Naur openjdk.orig/jdk/src/solaris/javavm/export/jvm_md.h openjdk/jdk/src/solaris/javavm/export/jvm_md.h
--- openjdk.orig/jdk/src/solaris/javavm/export/jvm_md.h	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/javavm/export/jvm_md.h	2020-07-11 11:28:15.215896637 +0200
@@ -65,7 +65,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
-#include <sys/signal.h>
+#include <signal.h>
 
 /* O Flags */
 
diff -Naur openjdk.orig/jdk/src/solaris/native/java/net/bsd_close.c openjdk/jdk/src/solaris/native/java/net/bsd_close.c
--- openjdk.orig/jdk/src/solaris/native/java/net/bsd_close.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/bsd_close.c	2020-07-11 11:28:49.367899002 +0200
@@ -38,7 +38,7 @@
 #include <sys/uio.h>
 #include <unistd.h>
 #include <errno.h>
-#include <sys/poll.h>
+#include <poll.h>
 
 /*
  * Stack allocated by thread when doing blocking operation
diff -Naur openjdk.orig/jdk/src/solaris/native/java/net/linux_close.c openjdk/jdk/src/solaris/native/java/net/linux_close.c
--- openjdk.orig/jdk/src/solaris/native/java/net/linux_close.c	2020-07-11 11:26:15.378888339 +0200
+++ openjdk/jdk/src/solaris/native/java/net/linux_close.c	2020-07-11 11:29:10.275900449 +0200
@@ -36,7 +36,7 @@
 #include <sys/uio.h>
 #include <unistd.h>
 #include <errno.h>
-#include <sys/poll.h>
+#include <poll.h>
 
 /*
  * Stack allocated by thread when doing blocking operation
diff -Naur openjdk.orig/jdk/src/solaris/native/java/net/net_util_md.h openjdk/jdk/src/solaris/native/java/net/net_util_md.h
--- openjdk.orig/jdk/src/solaris/native/java/net/net_util_md.h	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/native/java/net/net_util_md.h	2020-07-11 11:29:27.031901609 +0200
@@ -27,7 +27,7 @@
 #define NET_UTILS_MD_H
 
 #include <netdb.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/socket.h>
 
 int NET_Timeout(int s, long timeout);
diff -Naur openjdk.orig/jdk/src/solaris/native/sun/nio/ch/DevPollArrayWrapper.c openjdk/jdk/src/solaris/native/sun/nio/ch/DevPollArrayWrapper.c
--- openjdk.orig/jdk/src/solaris/native/sun/nio/ch/DevPollArrayWrapper.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/native/sun/nio/ch/DevPollArrayWrapper.c	2020-07-11 11:29:40.127902516 +0200
@@ -28,7 +28,7 @@
 #include "jvm.h"
 #include "jlong.h"
 #include "sun_nio_ch_DevPollArrayWrapper.h"
-#include <sys/poll.h>
+#include <poll.h>
 #include <unistd.h>
 #include <sys/time.h>
 
diff -Naur openjdk.orig/jdk/src/solaris/native/sun/nio/ch/NativeThread.c openjdk/jdk/src/solaris/native/sun/nio/ch/NativeThread.c
--- openjdk.orig/jdk/src/solaris/native/sun/nio/ch/NativeThread.c	2020-07-11 11:26:15.379888339 +0200
+++ openjdk/jdk/src/solaris/native/sun/nio/ch/NativeThread.c	2020-07-11 11:30:02.027904033 +0200
@@ -34,7 +34,7 @@
 
 #ifdef __linux__
   #include <pthread.h>
-  #include <sys/signal.h>
+  #include <signal.h>
   /* Also defined in net/linux_close.c */
   #define INTERRUPT_SIGNAL (SIGRTMAX - 2)
 #elif __solaris__
diff -Naur openjdk.orig/jdk/src/solaris/native/sun/nio/ch/Net.c openjdk/jdk/src/solaris/native/sun/nio/ch/Net.c
--- openjdk.orig/jdk/src/solaris/native/sun/nio/ch/Net.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/native/sun/nio/ch/Net.c	2020-07-11 11:30:16.005905001 +0200
@@ -23,7 +23,7 @@
  * questions.
  */
 
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <string.h>
diff -Naur openjdk.orig/jdk/src/solaris/native/sun/nio/fs/LinuxNativeDispatcher.c openjdk/jdk/src/solaris/native/sun/nio/fs/LinuxNativeDispatcher.c
--- openjdk.orig/jdk/src/solaris/native/sun/nio/fs/LinuxNativeDispatcher.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/native/sun/nio/fs/LinuxNativeDispatcher.c	2020-07-11 11:31:27.956909983 +0200
@@ -29,6 +29,7 @@
 #include "jlong.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <dlfcn.h>
 #include <errno.h>
diff -Naur openjdk.orig/jdk/src/solaris/native/sun/nio/fs/LinuxWatchService.c openjdk/jdk/src/solaris/native/sun/nio/fs/LinuxWatchService.c
--- openjdk.orig/jdk/src/solaris/native/sun/nio/fs/LinuxWatchService.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/native/sun/nio/fs/LinuxWatchService.c	2020-07-11 11:30:38.592906565 +0200
@@ -32,7 +32,7 @@
 #include <dlfcn.h>
 #include <sys/types.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/inotify.h>
 
 #include "sun_nio_fs_LinuxWatchService.h"
diff -Naur openjdk.orig/jdk/src/solaris/transport/socket/socket_md.c openjdk/jdk/src/solaris/transport/socket/socket_md.c
--- openjdk.orig/jdk/src/solaris/transport/socket/socket_md.c	2020-04-29 14:33:10.000000000 +0200
+++ openjdk/jdk/src/solaris/transport/socket/socket_md.c	2020-07-11 11:30:50.239907371 +0200
@@ -37,7 +37,7 @@
 #include <thread.h>
 #else
 #include <pthread.h>
-#include <sys/poll.h>
+#include <poll.h>
 #endif
 
 #include "socket_md.h"
