From 0a3f38b4f7199391437edf903bb803a90cd4f2f3 Mon Sep 17 00:00:00 2001
From: 
Date: Wed, 10 Apr 2019 01:31:16 -0500
Subject: [PATCH 2/2] use ascent, descent and height from XftFont *f instead of
 FT_Face face.

---
 src/rxvtfont.C | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/rxvtfont.C b/src/rxvtfont.C
index 90377a7..e80aa63 100644
--- a/src/rxvtfont.C
+++ b/src/rxvtfont.C
@@ -1239,11 +1239,22 @@ rxvt_font_xft::load (const rxvt_fontprop &prop, bool force_prop)
 
       FT_Face face = XftLockFace (f);
 
+/*
+ * use ascent, descent and height from XftFont *f instead of FT_Face face.
+ * this somehow reproduces the behaviour of the line height as seen on xterm.
+
       ascent  = (face->size->metrics.ascender + 63) >> 6;
       descent = (-face->size->metrics.descender + 63) >> 6;
       height  = max (ascent + descent, (face->size->metrics.height + 63) >> 6);
       width   = 0;
 
+ */
+
+      ascent  = f->ascent;
+      descent = f->descent;
+      height  = max (ascent + descent, f->height);
+      width   = 0;
+
       bool scalable = face->face_flags & FT_FACE_FLAG_SCALABLE;
 
       XftUnlockFace (f);
-- 
2.21.0

