From 3a0f53449431004b50125cef17803d4431c6affa Mon Sep 17 00:00:00 2001
From: 
Date: Wed, 10 Apr 2019 01:30:24 -0500
Subject: [PATCH 1/2] don't use g.width as a width of a character!

---
 src/rxvtfont.C | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/rxvtfont.C b/src/rxvtfont.C
index 1914539..90377a7 100644
--- a/src/rxvtfont.C
+++ b/src/rxvtfont.C
@@ -1267,12 +1267,21 @@ rxvt_font_xft::load (const rxvt_fontprop &prop, bool force_prop)
           XGlyphInfo g;
           XftTextExtents16 (disp, f, &ch, 1, &g);
 
+/*  
+ * bukind: don't use g.width as a width of a character!
+ * instead use g.xOff, see e.g.: http://keithp.com/~keithp/render/Xft.tutorial
+
           g.width -= g.x;
 
           int wcw = WCWIDTH (ch);
           if (wcw > 0) g.width = (g.width + wcw - 1) / wcw;
 
           if (width    < g.width       ) width    = g.width;
+ */
+          int wcw = WCWIDTH (ch);
+          if (wcw > 1) g.xOff = g.xOff / wcw;
+          if (width < g.xOff) width = g.xOff;
+
           if (height   < g.height      ) height   = g.height;
           if (glheight < g.height - g.y) glheight = g.height - g.y;
         }
-- 
2.21.0

