https://github.com/akrennmair/newsbeuter/issues/174

diff -ur a/newsbeuter-2.8/src/stflpp.cpp b/newsbeuter-2.8/src/stflpp.cpp
--- a/newsbeuter-2.8/src/stflpp.cpp	2014-01-19 01:10:41.000000000 -0200
+++ b/newsbeuter-2.8/src/stflpp.cpp	2015-04-09 23:49:37.768386550 -0200
@@ -16,7 +16,7 @@
  */
 
 stfl::form::form(const std::string& text) : f(0) {
-	ipool = stfl_ipool_create((std::string(nl_langinfo(CODESET)) + "//TRANSLIT").c_str());
+	ipool = stfl_ipool_create(nl_langinfo(CODESET));
 	if (!ipool) {
 		throw exception(errno);
 	}
@@ -82,7 +82,7 @@
 
 std::string stfl::quote(const std::string& text) {
 	scope_mutex lock(&quote_mtx);
-	stfl_ipool * ipool = stfl_ipool_create((std::string(nl_langinfo(CODESET)) + "//TRANSLIT").c_str());
+	stfl_ipool * ipool = stfl_ipool_create(nl_langinfo(CODESET));
 	std::string retval = stfl_ipool_fromwc(ipool,stfl_quote(stfl_ipool_towc(ipool,text.c_str())));
 	stfl_ipool_destroy(ipool);
 	return retval;
diff -ur a/newsbeuter-2.8/src/utils.cpp b/newsbeuter-2.8/src/utils.cpp
--- a/newsbeuter-2.8/src/utils.cpp	2014-01-19 01:10:41.000000000 -0200
+++ b/newsbeuter-2.8/src/utils.cpp	2015-04-09 23:47:39.570864222 -0200
@@ -262,7 +262,7 @@
 	if (strcasecmp(tocode.c_str(), fromcode.c_str())==0)
 		return text;
 
-	iconv_t cd = ::iconv_open((tocode + "//TRANSLIT").c_str(), fromcode.c_str());
+	iconv_t cd = ::iconv_open(tocode.c_str(), fromcode.c_str());
 
 	if (cd == reinterpret_cast<iconv_t>(-1))
 		return result;
@@ -493,7 +493,6 @@
 
 std::string utils::wstr2str(const std::wstring& wstr) {
 	std::string codeset = nl_langinfo(CODESET);
-	codeset.append("//TRANSLIT");
 	struct stfl_ipool * ipool = stfl_ipool_create(codeset.c_str());
 	std::string result = stfl_ipool_fromwc(ipool, wstr.c_str());
 	stfl_ipool_destroy(ipool);
