diff -Naurw gdb-7.11.orig/gdb/linux-nat.c gdb-7.11/gdb/linux-nat.c
--- gdb-7.11.orig/gdb/linux-nat.c	2016-02-09 19:19:39.000000000 -0800
+++ gdb-7.11/gdb/linux-nat.c	2016-02-24 15:34:29.115801748 -0800
@@ -186,6 +186,10 @@
 #define O_LARGEFILE 0
 #endif
 
+#ifndef W_STOPCODE
+#define W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
+
 /* Does the current host support PTRACE_GETREGSET?  */
 enum tribool have_ptrace_getregset = TRIBOOL_UNKNOWN;
 
@@ -4850,6 +4854,6 @@
   /* NPTL reserves the first two RT signals, but does not provide any
      way for the debugger to query the signal numbers - fortunately
      they don't change.  */
-  sigaddset (set, __SIGRTMIN);
-  sigaddset (set, __SIGRTMIN + 1);
+  sigaddset (set, SIGRTMIN);
+  sigaddset (set, SIGRTMIN + 1);
 }
