diff -Naur gdb-7.11.orig/gdb/gdbserver/target.h gdb-7.11/gdb/gdbserver/target.h
--- gdb-7.11.orig/gdb/gdbserver/target.h	2016-02-09 19:19:39.000000000 -0800
+++ gdb-7.11/gdb/gdbserver/target.h	2016-02-24 15:40:02.961881411 -0800
@@ -21,6 +21,7 @@
 #ifndef TARGET_H
 #define TARGET_H
 
+#include <sys/stat.h>
 #include <sys/types.h> /* for mode_t */
 #include "target/target.h"
 #include "target/resume.h"
